{% extends 'base.html' %}

{% load static %}

{% block content %}
<div class="main-content-wrapper">
    <main class="main-content">
        <div class="container-fluid">
            <h1>Immune Portal Experiment Output</h1>
            <div class="card">
                <div class="card-header">
                    Searchable Data (<b>Hover over columns to see description, click on rows to get further sample information</b>)
                </div>
                <div class="card-body">
		    <select id="urlVal" style="visibility:hidden;">
	    		<option value={{outputFilePath}} selected>Select genome </option>
		  </select>
		  <div id='forResultTable'>
		
		</div>
		<!--CREATING A POPUP SECTION AFTER USER CLICKS-->
		<div id="outer" style="display:none;">
			<div id="additionalInfo" style="display:none;float:left;">
				<br></br>
				<p style="margin-left:2px">Hidden table info:</p>
				
				<select onchange="miniTable()" name="colSelect" id="colSelect" style="display:none">
				
				</select>
				
				<br></br>
				
				
			</div>
			<div id="additionalInfo2" style="display:none;float:right;">
				
				
				
			</div>
		</div>
		<p>&nbsp;</p> 
		<div>
		
		<p id='gblabel'  style="display:none"><p>
		<div id='svgHolder' style="display:none;width=0px;height=0px;"></div>
		</div>
        </div>
            
            <!-- Adding a list of "Hello" greetings
            <div class="card">
                <div class="card-header">
                    Run "echo" for different languages
                </div>
                <div class="card-body">
                    <select id="greeting-select"></select>
                    <button id="run-button" class="btn btn-primary">Run</button>
                </div>
            </div>
            -->
	    <div class="card">
                <div class="card-header" >
                    Upset Plot (<b>Intersecting Selections of HLA by antigens</b>)
                </div>
                <div class="card-body" style="overflow:hidden" data-role="imagemagnifier" data-magnifier-mode="glass">
                    <img id="upset" width="1350" height="320" style="margin-left:-150px;" ></img>
                </div>
            </div>
            
            <div class="card">
                <div class="card-header" id="hlaLoadHeader">
                    
                </div>
                <div class="card-body">
                    <p>The red line indicates how the individual compares to 1000 genomes samples:</p>
                    <div id="hlaLoad"></div>
                    <!--<img  id="hlaload"></img>-->
                </div>
            </div>
            
            <div class="card" style='display:none'>
                <div class="card-header">
                    Individual and 1000 Genome HLA statistics (<b>Use the Dropdown to switch between sets</b>)
                </div>
                <div class="card-body">
                
                    <div id="myDiv2"></div>

                </div>
            </div>
            <div class="card">
                <div class="card-header" id="relGenExpHeader">
                    Gene Expression Relative to TCGA BRCA
                </div>
                <div class="card-body">
                <p>The red dot indicates how the individual's cancer sample compares to the TCGA gene expression for that cancer:</p>
                    <div id="boxes"></div>

                </div>
            </div>
            <div class="card">
                <div class="card-header">
                    Individual and 1000 Genome HLA statistics (<b>Use squares to the right of graph to switch between sets</b>)
                </div>
                <div class="card-body">
                <p>Comparing user presented HLAs with population HLAs:</p>
                    <div id="myDiv1"></div>

                </div>
            </div>
            
        </div>
    </main>
</div>
{% endblock content %}

{% block css %}
        <link rel="stylesheet" href="../../static/css/dataTables.bootstrap.min.css">
        <link rel="stylesheet" href="../../static/css/jquery-ui.css">
        <style>
       
        #resultTable th{
	  position:relative;
	  z-index:100;
	  
	}
	#resultTable td{
	  position:relative;
	  z-index:100;

	}

	.HlaWithComment{
	  position:relative;
	  z-index:100;
	}

	.HlaComment {
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99999;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:220px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.HlaWithComment:hover span.HlaComment{
	  visibility: visible;
	  opacity: 1;
	  width:190px;
	  position:absolute; 
	  z-index:99999;
	}
	
	.WtWithComment{
	cursor: pointer;
	  position:relative;
	  z-index:100;
	}

	.WtComment {
	cursor: pointer;
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99998;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:300px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.WtWithComment:hover span.WtComment{
	cursor: pointer;
	  visibility: visible;
	  opacity: 1;
	  width:300px;
	  position:absolute; 
	  z-index:99998;
	}
	</style>
  {% endblock %}
  
{% block scripts %}
<script src="{% static 'django_airavata_api/dist/airavata-api.js' %}"></script>


  <script src = "../../static/js/jquery.dataTables.min.js" defer ></script>
  <script src="../../static/js/dalliance-compiled.js" defer></script>
  <!--for IRNEO-->
  <script src="../../static/js/plotly-latest.min.js" ></script>
  <script src="../../static/js/d3.v4.js" ></script>
  <script src="../../static/js/jquery-1.12.4.js" ></script>
  <script src="../../static/js/jquery-ui.js" ></script>
  <script src="../../static/js/highcharts.js" ></script>
<script>

function miniTable() {
	console.log('selectValue',document.getElementById('colSelect').value);
	var ai2=document.getElementById('additionalInfo2');
	ai2.innerHTML="";
	var table = document.createElement('table');
	table.border="1";
	 var select=document.getElementById('colSelect');
	for (var i = 1; i < 2; i++){
	    var tr = document.createElement('tr');   

	    var td1 = document.createElement('td');
	    var td2 = document.createElement('td');
	    
	    //var tempText=sel.options[sel.selectedIndex].text;
	    var text1 = document.createTextNode(select.options[select.selectedIndex].text);
	    var text2 = document.createTextNode(select.value.toString());

	    td1.appendChild(text1);
	    td2.appendChild(text2);
	    tr.appendChild(td1);
	    tr.appendChild(td2);

	    table.appendChild(tr);
	}
	var p=document.createElement('p');
	p.innerHTML='description:';
	ai2.appendChild(p);
	ai2.appendChild(table);
	document.getElementById('additionalInfo2').style.display="block";
}

function CSVToArray(strData, strDelimiter) {
    // Check to see if the delimiter is defined. If not,
    // then default to comma.
    strDelimiter = (strDelimiter || "\t");
    // Create a regular expression to parse the CSV values.
    var objPattern = new RegExp((
    // Delimiters.
    "(\\" + strDelimiter + "|\\r?\\n|\\r|^)" +
    // Quoted fields.
    "(?:\"([^\"]*(?:\"\"[^\"]*)*)\"|" +
    // Standard fields.
    "([^\"\\" + strDelimiter + "\\r\\n]*))"), "gi");
    // Create an array to hold our data. Give the array
    // a default empty first row.
    var arrData = [[]];
    // Create an array to hold our individual pattern
    // matching groups.
    var arrMatches = null;
    // Keep looping over the regular expression matches
    // until we can no longer find a match.
    while (arrMatches = objPattern.exec(strData)) {
        // Get the delimiter that was found.
        var strMatchedDelimiter = arrMatches[1];
        // Check to see if the given delimiter has a length
        // (is not the start of string) and if it matches
        // field delimiter. If id does not, then we know
        // that this delimiter is a row delimiter.
        if (strMatchedDelimiter.length && (strMatchedDelimiter != strDelimiter)) {
            // Since we have reached a new row of data,
            // add an empty row to our data array.
            arrData.push([]);
        }
        // Now that we have our delimiter out of the way,
        // let's check to see which kind of value we
        // captured (quoted or unquoted).
        if (arrMatches[2]) {
            // We found a quoted value. When we capture
            // this value, unescape any double quotes.
            var strMatchedValue = arrMatches[2].replace(
            new RegExp("\"\"", "g"), "\"");
        } else {
            // We found a non-quoted value.
            var strMatchedValue = arrMatches[3];
        }
        // Now that we have our value string, let's add
        // it to the data array.
        arrData[arrData.length - 1].push(strMatchedValue);
    }
    // Return the parsed data.
    return (arrData.slice(0,arrData.length-1));
}




function CSV2JSON(csv) {
    var array = CSVToArray(csv);
    var objArray = [];
    
    
    for (var i = 1; i < array.length; i++) {
        objArray[i - 1] = {};
        for (var k = 0; k < array[0].length && k < array[i].length; k++) {
            var key = array[0][k];
            objArray[i - 1][key] = array[i][k]
        }
    }
    var addData={}
    addData["data"]=objArray;
    return (addData);
}





function transpose(a) {

  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }

  return t;
}

function numInstanceOrig(arr,index) {
  console.log('numinstance');
  console.log(arr);
  var a = [],
    b = [],
    prev;
  //console.log('arrlength');
  //console.log(arr[0].length);
  //arr.sort();
  for (var i = 0; i < arr[index].length; i++) {
    //if (arr[i] !== prev) {
      //var temp=String(arr[i]).split(",")[index];
      
      //console.log("postSplit");
      //console.log(temp2);
      console.log("index",index);
      //if(exclusive.includes(temp2)){
	      if (temp2== "NA"){
	      	//console.log("an NA");
	      	a.push("0.0");
	      }
	      else{
	      	a.push(temp2);
	      }
      //}
      
      }
    
  	b.push(0);
  console.log('apost');
  console.log(a);

  return [a, b];
}

function numInstance(arr,index,exclusive) {
  console.log('numinstance');
  console.log(arr);
  var a = [],
    b = [],
    prev;
  //console.log('arrlength');
  //console.log(arr[0].length);
  //arr.sort();
  for (var i = 0; i < arr[0].length; i++) {
    //if (arr[i] !== prev) {
      //var temp=String(arr[i]).split(",")[index];
      var t1=arr[0][i];
      //console.log('t1',t1);
      var temp=String(t1);
      //console.log("befSplit",temp);
      var temp2=temp.split(",")[index];
      //console.log("postSplit");
      //console.log(temp2);
      console.log("index",index);
      if(exclusive.includes(temp.split(",")[0])){
	      if (temp2== "NA"){
	      	//console.log("an NA");
	      	a.push("0.0");
	      }
	      else{
	      	a.push(temp2);
	      }
      }
      
      }
    
  	b.push(0);
  console.log('apost');
  console.log(a);

  return [a, b];
}


function transpose(a) {
  console.log('transpose');
  console.log(a);
  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }
  console.log('posttranspose');
  console.log(t);

  return t;
}






var a="{{request}}";
//console.log('a');
console.log('befdatafile',a);
b=a.split("?=")[1].split("&")[0];
console.log('datafile',b);
var data_file=b;
var data_file2=data_file.split("?")[1];



//var data_file2=data_file.split("?")[1];
//var imgName=a.split("?=")[1].split("&")[1];
//console.log('pname',imgName);


//$(document).ready(function() {
	//document.getElementById("seqlogo").src="";
	var dataout;
	var currcantype='';
	$.ajax({
	    url: 'https://immuneportal.ccbb.iupui.edu/api/experiments/'+data_file,
	    method: 'GET',
	    //datatype: 'json',
	    
	    
       	    //console.log(data_file);
            //console.log(data_file);
	    success: function (data) {
	    	//console.log('experiment info',data['experimentOutputs']);
	    	console.log(data);
	    	
	    	
	    	for(var i=0; i<data['experimentInputs'].length;i++){
	    		//console.log(Object.keys(data['experimentInputs'][i]))
	    		if(data['experimentInputs'][i]['name']=='Cancer Type'){
	    			currcantype=data['experimentInputs'][i]['value']
	    			console.log(currcantype);
	    		}
	    	}
	    	//SET THE CANCER TITLES ACCORDINGLY 
                document.getElementById('hlaLoadHeader').innerHTML="HLA Load (Load Based on Individual's <b>"+currcantype+"</b> Sample)"
                document.getElementById('relGenExpHeader').innerHTML="Gene Expression Relative to TCGA for cancer type <b>"+currcantype+"</b>"
                
	    	for(var i=0; i<data['experimentOutputs'].length;i++){
	    		if(data['experimentOutputs'][i]['name']=='Associated Files'){
	    			console.log('files',data['experimentOutputs'][i]['value']);
	    			var allFiles=data['experimentOutputs'][i]['value'].split(',');
	    			//console.log('#files',data['experimentOutputs'][i]['value'].split(',').length);
	    			///api/download?data-product-uri=airavata-dp%3A%2F%2Fddc9b4c5-7704-421a-a458-81a717203c6b
	    			
	    			var geneFreq='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[1];
	    			//var finalOutput='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[4];
	    			var tableData='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[0];
	    			
	    			
	    			//FOR GENEFREQ SECTION
	    			
	    			$.ajax({
				    url: geneFreq,
				    method: 'GET',
				    success: function (geneFreqData) {
	    				     //console.log(CSVToArray(geneFreqData))
					      
					      //TITLES OF CSV COLUMNS
						console.log(CSVToArray(geneFreqData)[0]);
						var temp=CSVToArray(geneFreqData);
						var geneFreqDataTitles=temp[0];
						var geneFreqDataArr=temp.slice(1, temp.length);
						var personalGenesDict={};
						console.log(geneFreqDataArr);
						for(var i=0;i<geneFreqDataArr.length;i++){
							

							personalGenesDict[geneFreqDataArr[i][0]]=parseFloat(geneFreqDataArr[i][1])
						}
						console.log('personGenes',personalGenesDict);
						$.ajax({
							    async: false,
							    url: 'https://immuneportal.ccbb.iupui.edu/static/files/OSF_TCGA_ALL_ICB_genes_exprs.csv',
							    method: 'GET',
							    //datatype: 'json',
							    
							    
						       	    //console.log(data_file);
							    //console.log(data_file);
							    success: function (tcgaGeneFreqs) {
							    		var temp=CSVToArray(tcgaGeneFreqs);
							    		//console.log(tcgaGeneFreqsArray);
									var tcgaGeneFreqsTitles=temp[0];
									console.log(tcgaGeneFreqsTitles);
									var tcgaGeneFreqsArr=temp.slice(1, temp.length);
									console.log(tcgaGeneFreqsArr[0]);
									
									var titleArr=tcgaGeneFreqsTitles[0].split(',');
									var geneDict={};
									//INITIALIZING GENE DICT
									for(var j=0;j<titleArr.length;j++){

										if(j==0| j==titleArr.length-1){
											continue;
										}
										else{
											geneDict[titleArr[j]]=[];
										}
										
									}
									//FILLING GENE DICT
									for(var i=0;i<tcgaGeneFreqsArr.length;i++){
										var currLst=tcgaGeneFreqsArr[i][0].split(',');
										if(currLst[titleArr.length-1]==currcantype){
											
											for(var j=0;j<titleArr.length;j++){

												if(j==0| j==titleArr.length-1){
													continue;
												}
												else{
													geneDict[titleArr[j]].push(parseFloat(currLst[j]));
												}
												
											}
										}
										else{
											continue;
										}
									}
									console.log('part of b2m',geneDict['B2M'].slice(0,5));
									
									var keysForIt=Object.keys(geneDict).reverse();
									var traces=[];
									for(var i=0;i<keysForIt.length;i++){
										var temptrace={
											x: geneDict[keysForIt[i]],
											marker: {color: '#000000'},
									  		type: 'box',
									  		name: keysForIt[i]
										}
										traces.push(temptrace);
										temptrace={
											x: [personalGenesDict[keysForIt[i]]],
											y: [keysForIt[i]],
											marker: {
												color: '#FF0000',
												/*line:{
													width:6
												}*/
											},
									  		type: 'scatter',
									  		//text: 'Patient '+keysForIt[i],
									  		name: 'Patient '+keysForIt[i]
										}
										traces.push(temptrace);
									}
							    		//TITLES OF CSV COLUMNS
							    		/*console.log(dataout[0]);
						    			var trace1 = {
									  x: [1, 2, 3, 4, 4, 4, 8, 9, 10],
									  type: 'box',
									  name: 'Set 1'
									};

									var trace2 = {
									  x: [2, 3, 3, 3, 3, 5, 6, 6, 7],
									  type: 'box',
									  name: 'Set 2'
									};*/

									var data = traces;

									var layout = {
									  title: 'Relative Gene Concentrations'
									};

									Plotly.newPlot('boxes', data, layout);
								}
						});
				
				
	    				}
	    			});
	    			//FOR GENEFREQ Section ENDED
	    			$.ajax({
				    async: false,
				    url: tableData,
				    method: 'GET',
				    //datatype: 'json',
				    
				    
			       	    //console.log(data_file);
				    //console.log(data_file);
				    success: function (data) {
				//console.log('hello');
						//console.log(data_file);
						    //console.log(data);
						    var dataout=CSVToArray(data);
						    dataoutTable = dataout.slice(1, dataout.length);
						    //console.log('postfunc',dataout);
						    //$.fn.dataTableExt.sErrMode = 'throw';
						    var base=document.getElementById('forResultTable');
						    var tbl=document.createElement('table');
						    tbl.id='resultTable'
						    tbl.classList.add('table');
						    tbl.classList.add('table-striped');
						    tbl.classList.add('table-bordered');
						    tbl.classList.add('table-hover');
						    
						    //tbl.set="table table-striped table-bordered table-hover";
						    tbl.style="cursor:pointer;"
						    tbl.cellspacing="0";
						    tbl.width="100%";
						    var thd=document.createElement('thead');
						    var tr=document.createElement('tr');
						    
						    var columnDefs=[]
						    console.log('DataOut',dataout[0]);
						    for(var i=0;i<dataout[0].length;i++){
						    	var th=document.createElement('th');
						    	if(dataout[0][i]!= 'peptide' & dataout[0][i]!='geneSymbol' & dataout[0][i]!='gene'){
						    		th.style="display:none";
						    		columnDefs.push({"targets":[i],"visible":false,"searchable":false});
						    	}
						    	//th.placeholder=dataout[0][i];
						    	//th.text=dataout[0][i];
						    	//th.abbr=dataout[0][i];
						    	th.innerHTML=dataout[0][i];
						    	tr.appendChild(th);
						    	
						    	
						    }
						    thd.appendChild(tr);
						    tbl.appendChild(thd);
						    base.appendChild(tbl);
						    
					/*	    
						    <table id="resultTable" class="table table-striped table-bordered table-hover" style="cursor:pointer;" cellspacing="0" width="100%">
				  <thead>
				      <tr>
					   <th style="display:none">id</th>
					    <th class="HlaWithComment">hla <span class="HlaComment"  >human leukocyte antigen (HLA) allele </span> </th>
					    
					    <th class="WtWithComment" >mut_pep<span class="WtComment">peptide derived from retained intron regions; </span></th>
					    <th style="display:none" class="WtWithComment">wt_pep <span class="WtComment">WildType Peptide: </span></th>
					    <th style="display:none">pep_hydro</th>
					    <th style="display:none">pep_polar</th>
					    <th style="display:none">pep_charge</th>
					    <th style="display:none">pep_entropy</th>
					    <th style="display:none">pep_molsize</th>
					    <th style="display:none">pep_TAP</th>
					    <th style="display:none">pep_Cle</th>
					    <th style="display:none">pep_Comb</th>
					    <th style="display:none">neo_aff</th>
					    <th style="display:none">neo_rank</th>
					    <th style="display:none">iedb_immunescore</th>
					    <th style="display:none">pep_DAI</th>
					    <th style="display:none">pep_Rscore</th>
					    <th style="display:none">pep_NRP</th>
					    <th style="display:none"> pep_pTuneos</th>
					    <th>rf_predict</th>
				      </tr>
				  </thead>
				</table>
				*/
						    
						    $('#resultTable').DataTable( {
						    		"data":dataoutTable,
						    		"deferRender": true,
						    		"scrollX":  200,
						    		"scrollY":  200,
						    		"scrollCollapse": true,
						    		"scroller": true,
						    		
						    		"columnDefs": columnDefs
								
							    } );
							    var table = $('#resultTable').DataTable();
								$('#resultTable').on('click', 'tbody tr', function() {
								  console.log('API row values : ', table.row(this).data());
								  var clickdata=table.row(this).data();
								  
								  
								  console.log(clickdata);
								  console.log(dataout[0]);
								  var availableTags=dataout[0];
								  
								
								    	  
								
							document.getElementById('additionalInfo').style.display="block";
							//document.getElementById('additionalInfo2').style.display="block";
							document.getElementById('outer').style.display="block";
							
							var select=document.getElementById('colSelect');
							for(var i=0;i<availableTags.length;i++){
								var tempOpt=document.createElement("option");
								console.log(availableTags[i]);
								tempOpt.value=clickdata[i];
								tempOpt.text=availableTags[i];
								select.appendChild(tempOpt);
							}
							
							document.getElementById('colSelect').style.display="block";
		

								  /*
								  document.getElementById('additionalInfo').innerHTML="<u><h6>Additional Info</h6></u><ul><li class='WtWithComment'>pep_hydro:"+clickdata[4]+"<span class='WtComment'>sum of amino acids hydrophobicity for peptide</span></li><li>pep_polar:"+clickdata[5]+"</li><li>pep_charge:"+clickdata[6]+"</li><li class='WtWithComment' >pep_entropy:"+clickdata[7]+"<span class='WtComment'>sum of amino acids residues entropy for peptide </span></li><li class='WtWithComment' >pep_molsize:"+clickdata[8]+"<span class='WtComment'>sum of amino acids residues molecular weight for peptide</span></li><li class='WtWithComment' >pep_TAP:"+clickdata[9]+"<span class='WtComment'>TAP transport efficiency, predicted by NetCTL1.2 </span></li><li class='WtWithComment' >pep_Cle:"+clickdata[10]+"<span class='WtComment'>C terminal cleavage affinity, predicted by NetCTL1.2 </span></li><li class='WtWithComment' >pep_Comb:"+clickdata[11]+"<span class='WtComment'>Overall prediction score, predicted by NetCTL1.2</span></li><li class='WtWithComment' >neo_aff:"+clickdata[12]+"<span class='WtComment'>NetMHCpan 4.1 predicted IC50 values (Aff, nM unit) </span></li><li class='WtWithComment' >neo_rank:"+clickdata[13]+"<span class='WtComment'>NetMHCpan 4.1 predicted percentile rank binding affinity value </span></li><li class='WtWithComment' >iedb_immunescore:"+clickdata[14]+"<span class='WtComment'>Class I Immunogenicity score from IEDB, http://tools.iedb.org/immunogenicity;</span></li><li class='WtWithComment' >pep_DAI:"+clickdata[15]+"<span class='WtComment'>differential agretopicity index, from Ghorani et al’s study (PMID:29361136); </span></li><li class='WtWithComment' >pep_Rscore:"+clickdata[16]+"<span class='WtComment'>neoantigen fitness score from Łuksza et al’s study (PMID:29132144); </span></li><li class='WtWithComment' >pep_NRP:"+clickdata[17]+"<span class='WtComment'>Neoantigen Recognition Potential score, Balachandran et al’s study (PMID:29132146); </span></li><li class='WtWithComment' >pep_pTuneos:"+clickdata[18]+"<span class='WtComment'>pTuneos prediction score, Zhou et al’s study (PMID:31666118); </span></li></ul> <u><h6>Embedded Genome location</h6></u>";*/
								  
								  //console.log(clickdata[1])
								  var temp=clickdata[1].split('@')[1]
								  var chr=temp.split(':')[0].substring(3)
								  var start=parseInt(temp.split(':')[1].split('-')[0])
								  var end=parseInt(temp.split(':')[1].split('-')[1])
								   browser.setLocation(chr, start-50, end + 50);
								   document.getElementById('gblabel').style.display="block";
								  document.getElementById('svgHolder').style.display="block";
								  
								})
								var browser =  new Browser({
								      chr:          '22',
								      viewStart:    30700000,
								      viewEnd:      30900000,

								      coordSystem: {
									speciesName: 'Human',
									taxon: 9606,
									auth: 'GRCh',
									version: '38',
									ucscName: 'hg38'
								      },

								      sources:     [{name:                 'Genome',
										     twoBitURI:            'https://www.biodalliance.org/datasets/hg19.2bit',
										     tier_type:            'sequence'},
										    {name:                 'Genes',
										     desc:                 'Gene structures from GENCODE 19',
										     bwgURI:               'https://www.biodalliance.org/datasets/gencode.bb',
										     stylesheet_uri:       'https://www.biodalliance.org/stylesheets/gencode.xml',
										     collapseSuperGroups:  true,
										     trixURI:              'https://www.biodalliance.org/datasets/geneIndex.ix'},
										    {name:                 'Repeats',
										     desc:                 'Repeat annotation from Ensembl',
										     bwgURI:               'https://www.biodalliance.org/datasets/repeats.bb',
										     stylesheet_uri:       'https://www.biodalliance.org/stylesheets/bb-repeats.xml'},
										    {name:                 'Conservation',
										     desc:                 'Conservation', 
										     bwgURI:               'https://immuneportal.ccbb.iupui.edu/data/hg19.100way.phyloP100way.bw',
										     noDownsample:         true}],

								    });

					    
					    //HLALOAD SECTION
					    $.ajax({
					    async: false,
					    url: 'https://immuneportal.ccbb.iupui.edu/static/files/TCGA_IRneoAg_load_in_8cancers_summary.csv',
					    method: 'GET',
					    //datatype: 'json',
					    
					    
					       	    //console.log(data_file);
						    //console.log(data_file);
						    success: function (TCGAbreakdown) {
						    		var TCGAarray=CSVToArray(TCGAbreakdown);
						    		console.log(TCGAarray);
						    		//var TCGAjson=CSV2JSON(TCGAbreakdown);
						    		//console.log('tcgajson',TCGAjson);
						    		var currCanc=[];
						    		for(var i=1;i<TCGAarray.length;i++){
						    			if(TCGAarray[i][0].includes(currcantype)){
						    				currCanc.push(parseInt(TCGAarray[i][0].split(',')[1]));
						    			}
						    		}
						    		console.log('currcanc',currCanc);
						    		console.log('CURRCANCE Min:',Math.min.apply(Math,currCanc));
						    		console.log('CURRCANCE RANGE:',Math.max.apply(Math,currCanc)-Math.min.apply(Math,currCanc));
						    		var currCancRange=Math.max.apply(Math,currCanc)-Math.min.apply(Math,currCanc)
						    		/*

								  var margin = { top: 30, right: 30, bottom: 30, left: 50 },
								    width = 500 - margin.left - margin.right,
								    height = 500 - margin.top - margin.bottom;

								  // append the svg object to the body of the page
								  var svg = d3.select("#my_dataviz")
								    .append("svg")
								    .attr("width", width + margin.left + margin.right)
								    .attr("height", height + margin.top + margin.bottom)
								    .append("g")
								    .attr("transform",
								      "translate(" + margin.left + "," + margin.top + ")");

								 var list= currCanc;
								  //var list = [10,10,11,11,12,12,12,15,15,16,16]

								  // add the x Axis
								  var x = d3.scaleLinear()
								    .domain([0,20000])
								    .range([0, width]);
								  svg.append("g")
								    .attr("transform", "translate(0," + height + ")")
								    .call(d3.axisBottom(x));

								  // add the y Axis
								  var y = d3.scaleLinear()
								    .range([0, height])
								    .domain([0, 0.0005]);
								  svg.append("g")
								    .call(d3.axisLeft(y));

								  // Compute kernel density estimation
								  var kde = kernelDensityEstimator(kernelEpanechnikov(7), x.ticks(40))
								  var density = kde(list)

								  // Plot the area
								  svg.append("path")
								    .attr("class", "mypath")
								    .datum(density)
								    .attr("fill", "#69b3a2")
								    .attr("opacity", ".8")
								    .attr("stroke", "#000")
								    .attr("stroke-width", 1)
								    .attr("stroke-linejoin", "round")
								    .attr("d", d3.line()
								      .curve(d3.curveBasis)
								      .x(function (d) { return x(d[0]); })
								      .y(function (d) { return y(d[1]); })
								    );
								    
								    
								    
								svg.append("line")
								.attr("x1", x(dataout.length-1))  //<<== change your code here
								.attr("y1", 0)
								.attr("x2", x(dataout.length-1))  //<<== and here
								.attr("y2", height - margin.top - margin.bottom)
								.style("stroke-width", 2)
								.style("stroke", "red")
								.style("fill", "none");
								
								    svg.append("text")
								    .attr("class", "x label")
								    .attr("text-anchor", "end")
								    .attr("x", width)
								    .attr("y", height - 6)
								    .text("NeoAntigen Load");
								    
								    
								    svg.append("text")
								    .attr("class", "y label")
								    .attr("text-anchor", "end")
								    .attr("y", 6)
								    .attr("dy", ".75em")
								    .attr("transform", "rotate(-90)")
								    .text("frequency");
								    



								  // Function to compute density
								  function kernelDensityEstimator(kernel, X) {
								    return function (V) {
								      return X.map(function (x) {
									return [x, d3.mean(V, function (v) { return kernel(x - v); })];
								      });
								    };
								  }
								  function kernelEpanechnikov(k) {
								    return function (v) {
								      return Math.abs(v /= k) <= 1 ? 0.75 * (1 - v * v) / k : 0;
								    };
								  }*/
								  
								  
								let dataSource = currCanc;
								let xiData = [];
								let animationDuration = 0;
								
								let range = currCancRange,
								  startPoint = currCanc[0];
								for (i = 0; i < range; i++) {
								  xiData[i] = startPoint + i;
								}
								let data = [];

								function GaussKDE(xi, x) {
								  return (1 / Math.sqrt(2 * Math.PI)) * Math.exp(Math.pow((xi - x), 2) / -2);
								}

								let N = dataSource.length;

								for (i = 0; i < xiData.length; i++) {
								  let temp = 0;
								  for (j = 0; j < dataSource.length; j++) {
								    temp = temp + GaussKDE(xiData[i], dataSource[j]);
								  }
								  data.push([xiData[i], (1 / N) * temp]);
								}

								Highcharts.chart("hlaLoad", {
								  chart: {
								    type: "line",
								    animation: true
								  },
								  title: {
								    text: "Gaussian Kernel Density Estimation (KDE)"
								  },
								  credits: {
    									enabled: false
  								  },
								  yAxis: {
								    title: { text: null }
								  },
								  tooltip: {
								    valueDecimals: 3
								  },
								  plotOptions: {
								    series: {
								      marker: {
									enabled: false
								      },
								      dashStyle: "shortdot",
								      color: "#ff8d1e",
								      pointStart: xiData[0],
								      animation: {
									duration: animationDuration
								      }
								    }
								  },
								  series: [
								    {
								      name: "KDE",
								      dashStyle: "solid",
								      lineWidth: 2,
								      color: "#1E90FF",
								      data: data
								    }
								  ]
								});  
								  
							}
					    });
					    
					    
					    
					    
					    
					    
					    
					    
					    $.ajax({
					   async: false,
					   type: "GET",
					    url: "https://immuneportal.ccbb.iupui.edu/static/files/1000genomes_HLA_freq_statistics.csv",
					    

					    success: function (data) {
					    	var genomes=CSVToArray(data);
					    	console.log('made it here');
					    	
						//console.log('json',data);
							var userspecific=dataout;
							console.log('userspecific',userspecific);
							console.log('indexof',userspecific[0].indexOf('hla'));
							console.log('transposedus',transpose(userspecific.slice(1,userspecific.length)));
							var allTypes=transpose(userspecific.slice(1,userspecific.length))[userspecific[0].indexOf('hla')]
							console.log('allTypes',allTypes);
							console.log('genomes',transpose(genomes.slice(1,genomes.length)));
							var count = {};
							allTypes.forEach(e => count[e] ? count[e]++ : count[e] = 1 );
							console.log("count");
							console.log(count);
							
							var nKeys=Object.values(count).length;
							
							var values=Object.values(count);
							console.log('thesevals',values);
							var proportions=new Array(nKeys);
							var total=Object.values(count).reduce((a, b) => a + b, 0);
							for (i=0; i<nKeys; i++){
							     proportions[i]=values[i]/total;
							}
							console.log("Individual HLA",Object.keys(count));
							//count.entries();
							//var hlatypes= numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('HLA_list'))[0];
							
							//console.log(transpose(genomes.slice(1,genomes.length)));
							//console.log('hlalist');
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length)),[genomes[0].indexOf('HLA_list')])[0]);
							//console.log('all');
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length)),[genomes[0].indexOf('ALL')])[0]);
							//console.log(genomes.slice(0))
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length))[genomes[0].indexOf('')])[0]
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length))[genomes[0].indexOf('')])[0]
							//var hlatypes= numInstanceOrig(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('HLA_list'))[0];
							
							//console.log("indexTest",String(genomes[0]).split(",").indexOf('HLA_list'))
							console.log('HOPEFULLY KEYS',Object.keys(count));
							console.log('numinstance',numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('SAS'),Object.keys(count)));
							
							/*var trace1 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('ALL'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'ALL' 
							};*/

							var trace2 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('AFR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'AFR' 
							};

							var trace3 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('AMR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'AMR' 
							};

							var trace4 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('EAS'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'EAS' 
							};
							var trace5 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('EUR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'EUR' 
							};

							var trace6 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('SAS'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'SAS' 
							};
							
							var trace7 = {
							  x: Object.keys(count),
							  y: proportions,
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'User Sample' 
							};

							var data = [ trace2, trace3, trace4,trace5,trace6,trace7];

							var layout = {
							  //grid: {rows: 1, columns: 2, pattern: 'independent'},
							  autosize: true,
							  title: {text:"1000 Genomes HLA Proportions"},
							  xaxis: {title: { text:'HLA subtype'}},
							  yaxis: {title: { text:'Proportion'}},
							  margin: {
							    l: 50,
							    r: 50,
							    b: 100,
							    t: 100,
							    pad: 4
							  },
										  
							  //width: 1800,
					 		  //height: 1250,
							};



							Plotly.newPlot('myDiv1', data, layout);
						
					    }
					  });
					    
					    
					  	
					}
				});
				
				

			
			
			
			$.ajax({
			   async: false,
			   type: "GET",
			   
			    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/upset_view",
			    contentType: "image/jpg",
			    data:tableData.split('=')[1],
			    success: function (data) {
				//console.log('json',data);
				console.log(data);
				console.log('made it');
				dataUris=data.split(',')
				//console.log('https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+data)
				document.getElementById('upset').src = 'data:image/jpg;base64,' + data;
			//document.getElementById('hlaload').src = 'data:image/jpg;base64,' + +dataUris[0];
				   
				
			    
				
			    }
			});
			
			
	
	
    
    //console.log('firstOnly',output[0]);
    //$.fn.dataTable.ext.errMode = 'throw';
    
    /*$.each(data, function(i, data) {
	    var body = "<tr>";
	    body    += "<td>" + data.id + "</td>";
	    body    += "<td>" + data.hla + "</td>";
	    body    += "<td>" + data.mut_pep + "</td>";
	    body    += "<td>" + data.wt_pep + "</td>";
	    body    += "<td>" + data.pep_hydro + "</td>";
	    body    += "<td>" + data.pep_polar + "</td>";
	    body    += "<td>" + data.pep_charge + "</td>";
	    body    += "<td>" + data.pep_entropy + "</td>";
	    body    += "<td>" + data.pep_molsize + "</td>";
	    body    += "<td>" + data.pep_TAP + "</td>";
	    body    += "<td>" + data.pep_Cle + "</td>";
	    body    += "<td>" + data.pep_Comb + "</td>";
	    body    += "<td>" + data.neo_aff + "</td>";
	    body    += "<td>" + data.neo_rank + "</td>";
	    body    += "<td>" + data.iedb_immunescore + "</td>";
	    body    += "<td>" + data.pep_DAI + "</td>";
	    body    += "<td>" + data.pep_Rscore + "</td>";
	    body    += "<td>" + data.pep_NRP + "</td>";
	    body    += "<td>" + data.pep_pTuneos + "</td>";
	    body    += "<td>" + data.rf_predict + "</td>";
	    
	    body    += "</tr>";
	    $( '#resultTable').append(body);
    });*/
    
    
    
    
    
  

	    		
	    		break;
	    		}
	    	}
	    }
	});
	//console.log('https://immuneportal.ccbb.iupui.edu/api/experiments/'+data_file);
	
	
	
	    
    
    //console.log('firstOnly',output[0]);
    //$.fn.dataTable.ext.errMode = 'throw';
    
    /*$.each(data, function(i, data) {
	    var body = "<tr>";
	    body    += "<td>" + data.id + "</td>";
	    body    += "<td>" + data.hla + "</td>";
	    body    += "<td>" + data.mut_pep + "</td>";
	    body    += "<td>" + data.wt_pep + "</td>";
	    body    += "<td>" + data.pep_hydro + "</td>";
	    body    += "<td>" + data.pep_polar + "</td>";
	    body    += "<td>" + data.pep_charge + "</td>";
	    body    += "<td>" + data.pep_entropy + "</td>";
	    body    += "<td>" + data.pep_molsize + "</td>";
	    body    += "<td>" + data.pep_TAP + "</td>";
	    body    += "<td>" + data.pep_Cle + "</td>";
	    body    += "<td>" + data.pep_Comb + "</td>";
	    body    += "<td>" + data.neo_aff + "</td>";
	    body    += "<td>" + data.neo_rank + "</td>";
	    body    += "<td>" + data.iedb_immunescore + "</td>";
	    body    += "<td>" + data.pep_DAI + "</td>";
	    body    += "<td>" + data.pep_Rscore + "</td>";
	    body    += "<td>" + data.pep_NRP + "</td>";
	    body    += "<td>" + data.pep_pTuneos + "</td>";
	    body    += "<td>" + data.rf_predict + "</td>";
	    
	    body    += "</tr>";
	    $( '#resultTable').append(body);
    });*/
    
    
    
    
	/*
    $.ajax({
	   type: "GET",
	    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/image_view/?"+data_file2,
	    contentType: "image/jpg",

	    success: function (data) {
		//console.log('json',data);
		console.log('made it');
		document.getElementById('seqlogo').src = 'data:image/jpg;base64,' + data;
	    }
	});
	*/
	
	
    //});

	
    
	
   
	    	
	    
/*

	    $.fn.dataTable.ext.errMode = 'throw';
	    $('#resultTable').DataTable( {
		    "processing": true,
		"scrollX": true,
		"data":output,
	    } );
	    var table = $('#resultTable').DataTable();*/
/*
var data = [{
  type: 'table',
  header: {
    values: output[0],
    align: "center",
    line: {width: 1, color: 'black'},
    fill: {color: "grey"},
    font: {family: "Arial", size: 12, color: "white"}
  },
  cells: {
    values: transpose(output.slice(1,output.length)),
    align: "center",
    line: {color: "black", width: 1},
    font: {family: "Arial", size: 11, color: ["black"]}
  }
}]

Plotly.newPlot('myPltTable', data);

//CLICK EVENTS FOR PLOT TO UPDATE TABLE

myPlot.on('plotly_click', function(data){
    console.log(data);
    
    var newData=[];
    var curveNumber=data.points[0].curveNumber;
    for (i=1;i<output.length;i++){
    	//probability curve
    	if(curveNumber==0){
    		if(output[i].includes(data.points[0].y)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==1){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==2){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    }
    
    var data = [{
	  type: 'table',
	  header: {
	    values: output[0],
	    align: "center",
	    line: {width: 1, color: 'black'},
	    fill: {color: "grey"},
	    font: {family: "Arial", size: 12, color: "white"}
	  },
	  cells: {
	    values: transpose(newData.slice(0,newData.length)),
	    align: "center",
	    line: {color: "black", width: 1},
	    font: {family: "Arial", size: 11, color: ["black"]}
	  }
	}]
    //document.getElementById("myPltTable").setAttribute('data',data);
    //Plotly.redraw('myPltTable');
    Plotly.newPlot('myPltTable', data);

    
    /*var pts = '';
    for(var i=0; i < data.points.length; i++){
        pts = 'x = '+data.points[i].x +'\ny = '+
            data.points[i].y.toPrecision(4) + '\n\n';
    }
    alert('Closest point clicked:\n\n'+pts);
});*/







</script>
<script>
    
    const { models, services, session, utils } = AiravataAPI;
    
    

    const appInterfaceId = "Echo_23d67491-1bef-47bd-a0f5-faf069e09773";

</script>

{% endblock scripts %}
