{% extends 'base.html' %}

{% load static %}

{% block content %}
<div class="main-content-wrapper">
    <main class="main-content">
        <div class="container-fluid">
            <h1>Immune Portal Experiment Output</h1>
            <div class="card">
                <div class="card-header">
                    Searchable Data (<b>Hover over columns to see description, click on rows to get further sample information</b>)
                </div>
                <div class="card-body">
		    <select id="urlVal" style="visibility:hidden;">
	    		<option value={{outputFilePath}} selected>Select genome </option>
		  </select>
		<table id="resultTable" class="table table-striped table-bordered table-hover" style="cursor:pointer;" cellspacing="0" width="100%">
		  <thead>
		      <tr>
		           <th>id</th>
			    <th class="HlaWithComment">hla <span class="HlaComment"  >HLA Subtype: </span> </th>
			    
			    <th>mut_pep</th>
			    <th class="WtWithComment">wt_pep <span class="WtComment">WildType Peptide: </span></th>
			    <th>pep_hydro</th>
			    <th>pep_polar</th>
			    <th>pep_charge</th>
			    <th>pep_entropy</th>
			    <th>pep_molsize</th>
			    <th>pep_TAP</th>
			    <th>pep_Cle</th>
			    <th>pep_Comb</th>
			    <th>neo_aff</th>
			    <th>neo_rank</th>
			    <th>iedb_immunescore</th>
			    <th>pep_DAI</th>
			    <th>pep_Rscore</th>
			    <th>pep_NRP</th>
			    <th>pep_pTuneos</th>
			    <th>rf_predict</th>
		      </tr>
		  </thead>
		</table>
		<!--CREATING A POPUP SECTION AFTER USER CLICKS-->
		<div id="additionalInfo" style="visibility:hidden;">
			
		</div>
		<div id='svgHolder' style="display:none;width=0px;height=0px;"></div>
        </div>
            
            <!-- Adding a list of "Hello" greetings
            <div class="card">
                <div class="card-header">
                    Run "echo" for different languages
                </div>
                <div class="card-body">
                    <select id="greeting-select"></select>
                    <button id="run-button" class="btn btn-primary">Run</button>
                </div>
            </div>
            -->
	    <div class="card">
                <div class="card-header">
                    Upset Plot
                </div>
                <div class="card-body">
                    <img src="" id="upset"></img>
                </div>
            </div>
            
            
            <div class="card">
                <div class="card-header">
                    SeqLogo of Mutpep
                </div>
                <div class="card-body">
                    <img src="" id="seqlogo"></img>
                </div>
            </div>
            
            <div class="card">
                <div class="card-header">
                    Experiments
                </div>
                <div class="card-body">
                    <button id="refresh-button" class="btn btn-secondary">Refresh</button>
                    <table class="table">
                        <thead>
                            <tr>
                                <th scope="col">Name</th>
                                <th scope="col">Application</th>
                                <th scope="col">Creation Time</th>
                                <th scope="col">Status</th>
                                <th scope="col">Output</th>
                            </tr>
                        </thead>
                        <tbody id="experiment-list">
                        </tbody>
                    </table>
                </div>
            </div>
            
        </div>
    </main>
</div>
{% endblock content %}

{% block css %}
        <link rel="stylesheet" href="../../static/css/dataTables.bootstrap.min.css">
        <style>
       
        #resultTable th{
	  position:relative;
	  z-index:100;
	  
	}
	#resultTable td{
	  position:relative;
	  z-index:100;

	}

	.HlaWithComment{
	  position:relative;
	  z-index:100;
	}

	.HlaComment {
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99999;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:220px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.HlaWithComment:hover span.HlaComment{
	  visibility: visible;
	  opacity: 1;
	  width:190px;
	  position:absolute; 
	  z-index:99999;
	}
	
	.WtWithComment{
	  position:relative;
	  z-index:100;
	}

	.WtComment {
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99998;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:300px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.WtWithComment:hover span.WtComment{
	  visibility: visible;
	  opacity: 1;
	  width:300px;
	  position:absolute; 
	  z-index:99998;
	}
	</style>
  {% endblock %}
  
{% block scripts %}
<script src="{% static 'django_airavata_api/dist/airavata-api.js' %}"></script>


  <script src = "../../static/js/jquery.dataTables.min.js" defer ></script>
  <script src="../../static/js/dalliance-compiled.js" defer></script>
  <!--for IRNEO-->
  <script src="../../static/js/plotly-latest.min.js" ></script>
  
<script>



function CSVToArray(strData, strDelimiter) {
    // Check to see if the delimiter is defined. If not,
    // then default to comma.
    strDelimiter = (strDelimiter || "\t");
    // Create a regular expression to parse the CSV values.
    var objPattern = new RegExp((
    // Delimiters.
    "(\\" + strDelimiter + "|\\r?\\n|\\r|^)" +
    // Quoted fields.
    "(?:\"([^\"]*(?:\"\"[^\"]*)*)\"|" +
    // Standard fields.
    "([^\"\\" + strDelimiter + "\\r\\n]*))"), "gi");
    // Create an array to hold our data. Give the array
    // a default empty first row.
    var arrData = [[]];
    // Create an array to hold our individual pattern
    // matching groups.
    var arrMatches = null;
    // Keep looping over the regular expression matches
    // until we can no longer find a match.
    while (arrMatches = objPattern.exec(strData)) {
        // Get the delimiter that was found.
        var strMatchedDelimiter = arrMatches[1];
        // Check to see if the given delimiter has a length
        // (is not the start of string) and if it matches
        // field delimiter. If id does not, then we know
        // that this delimiter is a row delimiter.
        if (strMatchedDelimiter.length && (strMatchedDelimiter != strDelimiter)) {
            // Since we have reached a new row of data,
            // add an empty row to our data array.
            arrData.push([]);
        }
        // Now that we have our delimiter out of the way,
        // let's check to see which kind of value we
        // captured (quoted or unquoted).
        if (arrMatches[2]) {
            // We found a quoted value. When we capture
            // this value, unescape any double quotes.
            var strMatchedValue = arrMatches[2].replace(
            new RegExp("\"\"", "g"), "\"");
        } else {
            // We found a non-quoted value.
            var strMatchedValue = arrMatches[3];
        }
        // Now that we have our value string, let's add
        // it to the data array.
        arrData[arrData.length - 1].push(strMatchedValue);
    }
    // Return the parsed data.
    return (arrData.slice(0,arrData.length-1));
}




function CSV2JSON(csv) {
    var array = CSVToArray(csv);
    var objArray = [];
    
    
    for (var i = 1; i < array.length; i++) {
        objArray[i - 1] = {};
        for (var k = 0; k < array[0].length && k < array[i].length; k++) {
            var key = array[0][k];
            objArray[i - 1][key] = array[i][k]
        }
    }
    var addData={}
    addData["data"]=objArray;
    return (addData);
}





function transpose(a) {

  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }

  return t;
}


function numInstance(arr) {
  var a = [],
    b = [],
    prev;

  arr.sort();
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] !== prev) {
      a.push(arr[i]);
      b.push(1);
    } else {
      b[b.length - 1]++;
    }
    prev = arr[i];
  }

  return [a, b];
}


function transpose(a) {

  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }

  return t;
}


function numInstance(arr) {
  var a = [],
    b = [],
    prev;

  arr.sort();
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] !== prev) {
      a.push(arr[i]);
      b.push(1);
    } else {
      b[b.length - 1]++;
    }
    prev = arr[i];
  }

  return [a, b];
}



var a="{{request}}";
//console.log('a');
console.log('befdatafile',a);
b=a.split("?=")[1].split("&")[0];
console.log('datafile',b);
var data_file=b;


var data_file2=data_file.split("?")[1];
//var imgName=a.split("?=")[1].split("&")[1];
//console.log('pname',imgName);


$(document).ready(function() {
	document.getElementById("seqlogo").src="";
	var dataout;
	
	$.ajax({
	    url: data_file,
	    method: 'GET',
	    //datatype: 'json',
	    
	    
       	    //console.log(data_file);
            //console.log(data_file);
	    success: function (data) {
	//console.log('hello');
	//console.log(data_file);
	    //console.log(data);
	    var dataout=CSVToArray(data);
	    dataoutTable = dataout.slice(1, dataout.length);
	    //console.log('postfunc',dataout);
	    //$.fn.dataTableExt.sErrMode = 'throw';
	    $('#resultTable').DataTable( {
	    		"data":dataoutTable,
            		"deferRender": true,
            		"scrollX":  200,
            		"scrollY":  200,
            		"scrollCollapse": true,
            		"scroller": true,
			
		    } );
		    var table = $('#resultTable').DataTable();
			$('#resultTable').on('click', 'tbody tr', function() {
			  console.log('API row values : ', table.row(this).data());
			  var clickdata=table.row(this).data();
			  document.getElementById('additionalInfo').style.visibility="visible";
			  document.getElementById('additionalInfo').innerHTML="<u><h6>Additional Info</h6></u><ul><li>pep_hydro:"+clickdata[4]+"</li><li>pep_polar:"+clickdata[5]+"</li><li>pep_charge:"+clickdata[6]+"</li></ul> <u><h6>Embedded Genome location</h6></u>";
			  document.getElementById('svgHolder').style.display="block";
			  
			})
			var browser =  new Browser({
			      chr:          '22',
			      viewStart:    30700000,
			      viewEnd:      30900000,

			      coordSystem: {
				speciesName: 'Human',
				taxon: 9606,
				auth: 'GRCh',
				version: '37',
				ucscName: 'hg19'
			      },

			      sources:     [{name:                 'Genome',
					     twoBitURI:            '//www.biodalliance.org/datasets/hg19.2bit',
					     tier_type:            'sequence'},
					    {name:                 'Genes',
					     desc:                 'Gene structures from GENCODE 19',
					     bwgURI:               '//www.biodalliance.org/datasets/gencode.bb',
					     stylesheet_uri:       '//www.biodalliance.org/stylesheets/gencode.xml',
					     collapseSuperGroups:  true,
					     trixURI:              '//www.biodalliance.org/datasets/geneIndex.ix'},
					    {name:                 'Repeats',
					     desc:                 'Repeat annotation from Ensembl',
					     bwgURI:               '//www.biodalliance.org/datasets/repeats.bb',
					     stylesheet_uri:       '//www.biodalliance.org/stylesheets/bb-repeats.xml'},
					    {name:                 'Conservation',
					     desc:                 'Conservation', 
					     bwgURI:               '//regsnps-intron.ccbb.iupui.edu/annotation/hg19.100way.phyloP100way.bw',
					     noDownsample:         true}],

			    });

		    
		    
		    
		  	function makeTrace(i) {
    //console.log('exists',numInstance(transpose(dataout.slice(1,dataout.length))[dataout[0].indexOf('hla')]));
			    if (i==dataout[0].indexOf('hla')){
			    return {
			    	type: 'pie',
			    	title: 'HLA Type Distribution',
				labels:numInstance(transpose(dataout.slice(1,dataout.length))[dataout[0].indexOf('hla')])[0],
				values:numInstance(transpose(dataout.slice(1,dataout.length))[dataout[0].indexOf('hla')])[1],
				
				visible: i === 1,
			       
			      

			    }
			   }
			  
			}
			//console.log('json',dataout);         
				    
			var myPlot= document.getElementById('myDiv');

			//Plotly.newPlot('myDiv', data, layout);


			Plotly.newPlot('myDiv', [dataout[0].indexOf('hla')].map(makeTrace), {
			    updatemenus: [{
				
				
				yanchor: 'top',
				buttons: [{
				    method: 'update',
				    args: ['visible', [true, false, false]],
				    label: 'HLA type'
				}]
			    }],
			});
		}
	});
	    
    
    //console.log('firstOnly',output[0]);
    //$.fn.dataTable.ext.errMode = 'throw';
    
    /*$.each(data, function(i, data) {
	    var body = "<tr>";
	    body    += "<td>" + data.id + "</td>";
	    body    += "<td>" + data.hla + "</td>";
	    body    += "<td>" + data.mut_pep + "</td>";
	    body    += "<td>" + data.wt_pep + "</td>";
	    body    += "<td>" + data.pep_hydro + "</td>";
	    body    += "<td>" + data.pep_polar + "</td>";
	    body    += "<td>" + data.pep_charge + "</td>";
	    body    += "<td>" + data.pep_entropy + "</td>";
	    body    += "<td>" + data.pep_molsize + "</td>";
	    body    += "<td>" + data.pep_TAP + "</td>";
	    body    += "<td>" + data.pep_Cle + "</td>";
	    body    += "<td>" + data.pep_Comb + "</td>";
	    body    += "<td>" + data.neo_aff + "</td>";
	    body    += "<td>" + data.neo_rank + "</td>";
	    body    += "<td>" + data.iedb_immunescore + "</td>";
	    body    += "<td>" + data.pep_DAI + "</td>";
	    body    += "<td>" + data.pep_Rscore + "</td>";
	    body    += "<td>" + data.pep_NRP + "</td>";
	    body    += "<td>" + data.pep_pTuneos + "</td>";
	    body    += "<td>" + data.rf_predict + "</td>";
	    
	    body    += "</tr>";
	    $( '#resultTable').append(body);
    });*/
    
    
    
    $.ajax({
	   type: "GET",
	    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/upset_view/?"+data_file2,
	    contentType: "image/jpg",

	    success: function (data) {
		//console.log('json',data);
		console.log('made it');
		document.getElementById('upset').src = 'data:image/jpg;base64,' + data;
	    }
	});
    $.ajax({
	   type: "GET",
	    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/image_view/?"+data_file2,
	    contentType: "image/jpg",

	    success: function (data) {
		//console.log('json',data);
		console.log('made it');
		document.getElementById('seqlogo').src = 'data:image/jpg;base64,' + data;
	    }
	});
	
	
    });

	
    
	
   
	    	
	    
/*

	    $.fn.dataTable.ext.errMode = 'throw';
	    $('#resultTable').DataTable( {
		    "processing": true,
		"scrollX": true,
		"data":output,
	    } );
	    var table = $('#resultTable').DataTable();*/
/*
var data = [{
  type: 'table',
  header: {
    values: output[0],
    align: "center",
    line: {width: 1, color: 'black'},
    fill: {color: "grey"},
    font: {family: "Arial", size: 12, color: "white"}
  },
  cells: {
    values: transpose(output.slice(1,output.length)),
    align: "center",
    line: {color: "black", width: 1},
    font: {family: "Arial", size: 11, color: ["black"]}
  }
}]

Plotly.newPlot('myPltTable', data);

//CLICK EVENTS FOR PLOT TO UPDATE TABLE

myPlot.on('plotly_click', function(data){
    console.log(data);
    
    var newData=[];
    var curveNumber=data.points[0].curveNumber;
    for (i=1;i<output.length;i++){
    	//probability curve
    	if(curveNumber==0){
    		if(output[i].includes(data.points[0].y)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==1){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==2){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    }
    
    var data = [{
	  type: 'table',
	  header: {
	    values: output[0],
	    align: "center",
	    line: {width: 1, color: 'black'},
	    fill: {color: "grey"},
	    font: {family: "Arial", size: 12, color: "white"}
	  },
	  cells: {
	    values: transpose(newData.slice(0,newData.length)),
	    align: "center",
	    line: {color: "black", width: 1},
	    font: {family: "Arial", size: 11, color: ["black"]}
	  }
	}]
    //document.getElementById("myPltTable").setAttribute('data',data);
    //Plotly.redraw('myPltTable');
    Plotly.newPlot('myPltTable', data);

    
    /*var pts = '';
    for(var i=0; i < data.points.length; i++){
        pts = 'x = '+data.points[i].x +'\ny = '+
            data.points[i].y.toPrecision(4) + '\n\n';
    }
    alert('Closest point clicked:\n\n'+pts);
});*/







</script>
<script>
    
    const { models, services, session, utils } = AiravataAPI;
    
    

    const appInterfaceId = "Echo_23d67491-1bef-47bd-a0f5-faf069e09773";

</script>

{% endblock scripts %}
