{% extends 'base.html' %}

{% load static %}

{% block content %}


<div class="main-content-wrapper">
    <main class="main-content">
        <div class="container-fluid">
            <h1>Immune Portal Experiment Output</h1>
            <div class="card">
                <div class="card-header">
                    Searchable Data (<b>Hover over columns to see description, click on rows to get further sample information/genome location</b>)
                </div>
                <div class="card-body">
		    <select id="urlVal" style="display:none;">
	    		<option value={{outputFilePath}} selected>Select genome </option>
		  </select>
		  <!--<div id='forResultTable'></div>-->
		  
		<span><select id="colShow" >
		    <option value="">Show/Hide Column</option>
		    <option value="peptide">peptide</option>
<option value="gene">gene</option>
<option value="tpm">tpm</option>
<option value="rf_predict_prob">rf_predict_prob</option>
<option value="ensembl">ensembl</option>
<option value="geneSymbol">geneSymbol</option>
<option value="HLA-A*24:02">HLA-A*24:02</option>
<option value="HLA-A*25:01">HLA-A*25:01</option>
<option value="HLA-B*07:02">HLA-B*07:02</option>
<option value="HLA-B*18:01">HLA-B*18:01</option>
<option value="HLA-C*07:02">HLA-C*07:02</option>
<option value="HLA-C*12:03">HLA-C*12:03</option>
<option value="hla">hla</option>
<option value="wt_pep">wt_pep</option>
<option value="pep_hydro">pep_hydro</option>
<option value="pep_polar">pep_polar</option>
<option value="pep_charge">pep_charge</option>
<option value="pep_entropy">pep_entropy</option>
<option value="pep_molsize">pep_molsize</option>
<option value="pep_TAP">pep_TAP</option>
<option value="pep_Cle">pep_Cle</option>
<option value="pep_Comb">pep_Comb</option>
<option value="neo_aff">neo_aff</option>
<option value="neo_rank">neo_rank</option>
<option value="iedb_immunescore">iedb_immunescore</option>
<option value="pep_DAI">pep_DAI</option>
<option value="pep_Rscore">pep_Rscore</option>
<option value="pep_NRP">pep_NRP</option>
<option value="pep_pTuneos">pep_pTuneos</option>
<option value="rf_predict">rf_predict</option>
<option value="pep_pTuneos">pep_pTuneos</option>>
		</select></span>

<span><button class="btn" id="download-csv">Download CSV</button></span>
<p id='colDescription'></p>

<div id="example-table"></div>



		<span style="color:#102D4F;font-size:12px;font-family:Arial, Helvetica, sans-serif">
  Showing <span id="search_count"></span> results from (<span id="total_count"></span>)  
</span>
<div id='rowData' style="display:none"><br/><p>All Row Columns</p></div>
<div id="example-table3" style="display:none"></div>
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		<p>&nbsp;</p> 
		<div>
		
		<p id='gblabel'  style="display:none"><p>
		<div id='svgHolder' style="display:none;width=0px;height=0px;"></div>
		</div>
        </div>
            
            <!-- Adding a list of "Hello" greetings
            <div class="card">
                <div class="card-header">
                    Run "echo" for different languages
                </div>
                <div class="card-body">
                    <select id="greeting-select"></select>
                    <button id="run-button" class="btn btn-primary">Run</button>
                </div>
            </div>
            -->
            
	    <div class="card">
                <div class="card-header" >
                    Upset Plot, Intersecting Selections of HLA by antigens (<b>click to see enlarged</b>):
                </div>
                <div class="card-body" style="overflow:hidden" >
		        
		            		<img id="myImg" class="maglarge" src='https://immuneportal.ccbb.iupui.edu/static/files/Upset.png' width="1350" height="320" style="margin-left:-150px;" ></img>
		            	
		        <div id="myModal" class="modal">
				  <span class="close">&times;</span>
				  <img class="modal-content" id="img01">
				  <div id="caption"></div>
				</div>
                </div>
            </div>
            
            <div class="card">
                <div class="card-header" id="hlaLoadHeader">
                </div>
                <div class="card-body">
                    <p>The red line indicates how the individual compares to TCGA samples:</p>
                    <img id="hlaLoad2" src='https://immuneportal.ccbb.iupui.edu/static/files/HLALoad.png'></img>
                    <!--<img  id="hlaload"></img>-->
                </div>
            </div>
            
            <div class="card" style='display:none'>
                <div class="card-header">
                    Individual and 1000 Genome HLA statistics (<b>Use the Dropdown to switch between sets</b>)
                </div>
                <div class="card-body">
                
                    <div id="myDiv2"></div>

                </div>
            </div>
            <div class="card">
                <div class="card-header" id="relGenExpHeader">
                    Gene Expression Relative to TCGA BRCA
                </div>
                <div class="card-body">
                <p>The red dot indicates how the individual's cancer sample compares to the TCGA gene expression for that cancer:</p>
                    <div id="boxes"></div>

                </div>
            </div>
            <div class="card">
                <div class="card-header">
                    Individual and 1000 Genome HLA statistics (<b>Use squares to the right of graph to switch between sets</b>)
                </div>
                <div class="card-body">
                <p>Relative proportion of population HLAs to those present in sample:</p>
                    <div id="myDiv1"></div>

                </div>
            </div>
            
        </div>
    </main>
</div>
{% endblock content %}

{% block css %}
        <link rel="stylesheet" href="../../static/css/dataTables.bootstrap.min.css">
        <link rel="stylesheet" href="../../static/css/fixedHeader.dataTables.min.css">
        <link rel="stylesheet" href="../../static/css/jquery-ui.css">
        <link rel="stylesheet" href="../../static/css/buttons.dataTables.min.css">
        <link href="../../static/css/tabulator.min.css" rel="stylesheet">
         
        <style>
	       
        #resultTable th{
	  position:relative;
	  z-index:100;
	  
	}
	#resultTable td{
	  position:relative;
	  z-index:100;

	}

	.HlaWithComment{
	  position:relative;
	  z-index:100;
	}

	.HlaComment {
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99999;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:220px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.HlaWithComment:hover span.HlaComment{
	  visibility: visible;
	  opacity: 1;
	  width:220px;
	  position:absolute; 
	  z-index:99999;
	}
	
	.WtWithComment{
	cursor: pointer;
	  position:relative;
	  z-index:100;
	}

	.WtComment {
	cursor: pointer;
	  visibility: hidden;
	  position:absolute; 
	  z-index: 99998;
	  border:1px;
	  background-color:white;
	  border-style:solid;
	  border-color:black;
	  width:300px;
	  color:black; 
	  top:2px; 
	  left:40px;
	}

	.WtWithComment:hover span.WtComment{
	cursor: pointer;
	  visibility: visible;
	  opacity: 1;
	  width:300px;
	  position:absolute; 
	  z-index:99998;
	}
	
	#myImg {
	  border-radius: 5px;
	  cursor: pointer;
	  transition: 0.3s;
	}

	#myImg:hover {opacity: 0.7;}

	/* The Modal (background) */
	.modal {
	  display: none; /* Hidden by default */
	  position: fixed; /* Stay in place */
	  z-index: 1; /* Sit on top */
	  padding-top: 100px; /* Location of the box */
	  left: 0;
	  top: 0;
	  width: 100%; /* Full width */
	  height: 100%; /* Full height */
	  overflow: auto; /* Enable scroll if needed */
	  background-color: rgb(0,0,0); /* Fallback color */
	  background-color: rgba(0,0,0,0.9); /* Black w/ opacity */
	}

	/* Modal Content (image) */
	.modal-content {
	  
	  display: block;
	  width: 200%;
	  
	}

	/* Caption of Modal Image */
	#caption {
	  margin: auto;
	  display: block;
	  width: 80%;
	  max-width: 700px;
	  text-align: center;
	  color: #ccc;
	  padding: 10px 0;
	  height: 150px;
	}

	/* Add Animation */
	.modal-content, #caption {  
	  -webkit-animation-name: zoom;
	  -webkit-animation-duration: 0.6s;
	  animation-name: zoom;
	  animation-duration: 0.6s;
	}

	@-webkit-keyframes zoom {
	  from {-webkit-transform:scale(0)} 
	  to {-webkit-transform:scale(1)}
	}

	@keyframes zoom {
	  from {transform:scale(0)} 
	  to {transform:scale(1)}
	}

	/* The Close Button */
	.close {
	  position: absolute;
	  top: 15px;
	  right: 35px;
	  color: #f1f1f1;
	  font-size: 40px;
	  font-weight: bold;
	  transition: 0.3s;
	}

	.close:hover,
	.close:focus {
	  color: #bbb;
	  text-decoration: none;
	  cursor: pointer;
	}

	/* 100% Image Width on Smaller Screens */
	@media only screen and (max-width: 700px){
	  .modal-content {
	    width: 100%;
	  }
	}
	.slider {
  -webkit-appearance: none;
  width: 70px;
  height: 1px;
  background: #000000;
  outline: none;
  opacity: 0.7;
  -webkit-transition: .2s;
  transition: opacity .2s;
}

.slider:hover {
  opacity: 1;
}

.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 1px;
  height: 20px;
  background: #000000;
  cursor: pointer;
}

.slider::-moz-range-thumb {
  width: 1px;
  height: 20px;
  background: #000000;
  cursor: pointer;
}

.sliderinput {
    
    margin: 0px;
    position: absolute;
    left: 72px;
}

.minbox {
    
    margin: 0px;
    position: absolute;
    left: 0px;
}
.maxbox {
    
    margin: 0px;
    position: absolute;
    left: 194px;
}

.sliderlabel {
    display: inline-block;
    margin-top: 100px;
}

#start {
    width: 120px;
    z-index: 2;
}

#end {
    width: 120px;
    z-index: 2;
}

.tabulator-footer {
    display: none;
}
	

	</style>
  {% endblock %}
  
{% block scripts %}
<script src="{% static 'django_airavata_api/dist/airavata-api.js' %}"></script>

  <script src="../../static/js/jquery-3.5.1.js" defer ></script>
  <script src = "../../static/js/jquery.dataTables.min.js" defer ></script>
  <script src="../../static/js/dataTables.fixedHeader.min.js" defer ></script>
  <script src="../../static/js/dataTables.buttons.min.js"defer ></script>
  <script src="../../static/js/buttons.colVis.min.js"defer ></script>
  <script src="../../static/js/jquery-ui.js" defer></script>
  <script src="../../static/js/tabulator.min.js" defer></script>
  
  <script src="../../static/js/dalliance-compiled.js" defer></script>
  <!--for IRNEO-->
  <script src="../../static/js/plotly-latest.min.js" ></script>
  <script src="../../static/js/d3.v4.js" ></script>
  <script src="../../static/js/highcharts.js" ></script>
  <script src="../../static/js/highcharts-more.js" ></script>
  <script src="../../static/js/data.js" ></script>
  <script src="../../static/js/jstat.min.js" ></script>
  <script src="../../static/js/processDensity.js" ></script>
  
  
  
 

  <script src="../../static/js/blowup.min.js"></script>

<script>


var modal = document.getElementById("myModal");

// Get the image and insert it inside the modal - use its "alt" text as a caption
var img = document.getElementById("myImg");
var modalImg = document.getElementById("img01");
var captionText = document.getElementById("caption");
img.onclick = function(){
  modal.style.display = "block";
  modalImg.src = this.src;
  captionText.innerHTML = this.alt;
}

// Get the <span> element that closes the modal
var span = document.getElementsByClassName("close")[0];

// When the user clicks on <span> (x), close the modal
span.onclick = function() { 
  modal.style.display = "none";
}









function miniTable() {


	descriptionDict={'Id': 'random assigned peptide id','hla':  'human leukocyte antigen (HLA) allele','mut_pep': 'peptide derived from retained intron regions','wt_pep': 'the most similar peptide from human reference protein database','pep_hydro': 'sum of amino acids hydrophobicity for peptide','pep_polar': 'sum of amino acids residues polarity for peptide','pep_charge': 'sum of amino acids residues net charge for peptide (at neutral pH)','pep_entropy': 'sum of amino acids residues entropy for peptide','pep_molsize': 'sum of amino acids residues molecular weight for peptide','pep_TAP':  'TAP transport efficiency, predicted by NetCTL1.2','pep_Cle': 'C terminal cleavage affinity, predicted by NetCTL1.2','pep_Comb': 'Overall prediction score, predicted by NetCTL1.2','neo_aff': 'NetMHCpan 4.1 predicted IC50 values (Aff, nM unit)','neo_rank': 'NetMHCpan 4.1 predicted percentile rank binding affinity value','iedb_immunescore': 'Class I Immunogenicity score from IEDB, http://tools.iedb.org/immunogenicity','pep_DAI': 'differential agretopicity index, from Ghorani et al’s study (PMID:29361136)','pep_Rscore': 'neoantigen fitness score from Łuksza et al’s study (PMID:29132144)','pep_NRP': 'Neoantigen Recognition Potential score, Balachandran et al’s study (PMID:29132146)','pep_pTuneos': 'pTuneos prediction score, Zhou et al’s study (PMID:31666118)','rf_predict': 'intronNeoantigen random forest model prediction, Positive/Negative represent the ability for recognition by CD8+ cytotoxic T cells' };
	console.log('selectValue',document.getElementById('colSelect').value);
	var ai2=document.getElementById('additionalInfo2');
	ai2.innerHTML="";
	var table = document.createElement('table');
	//table.border="1";
	 var select=document.getElementById('colSelect');
	for (var i = 1; i < 2; i++){
	    var tr = document.createElement('tr');   

	    var td1 = document.createElement('td');
	    var td2 = document.createElement('td');
	    
	    //var tempText=sel.options[sel.selectedIndex].text;
	    var text1 = document.createTextNode(select.options[select.selectedIndex].text);
	    var text2 = document.createTextNode(select.value.toString());
	    
	    td1.style="padding-left:5px;padding-right:5px";
	    td2.style="padding-left:5px;padding-right:5px";

	    td1.innerHTML='<b>'+select.options[select.selectedIndex].text+':</b>'
	    td2.innerHTML=select.value.toString()
	    //td1.appendChild(text1);
	    //td2.appendChild(text2);
	    tr.appendChild(td1);
	    tr.appendChild(td2);

	    table.appendChild(tr);
	}
	var p=document.createElement('p');
	p.innerHTML='Description: '+descriptionDict[select.options[select.selectedIndex].text];
	ai2.appendChild(p);
	table.style="border:solid black 1px;border-radius:6px"
	var br = document.createElement('br');
	ai2.appendChild(table);
	ai2.appendChild(br);
	ai2.appendChild(br);
	
	
	document.getElementById('additionalInfo2').style.display="block";
}

function CSVToArray(strData, strDelimiter) {
    // Check to see if the delimiter is defined. If not,
    // then default to comma.
    strDelimiter = (strDelimiter || "\t");
    // Create a regular expression to parse the CSV values.
    var objPattern = new RegExp((
    // Delimiters.
    "(\\" + strDelimiter + "|\\r?\\n|\\r|^)" +
    // Quoted fields.
    "(?:\"([^\"]*(?:\"\"[^\"]*)*)\"|" +
    // Standard fields.
    "([^\"\\" + strDelimiter + "\\r\\n]*))"), "gi");
    // Create an array to hold our data. Give the array
    // a default empty first row.
    var arrData = [[]];
    // Create an array to hold our individual pattern
    // matching groups.
    var arrMatches = null;
    // Keep looping over the regular expression matches
    // until we can no longer find a match.
    while (arrMatches = objPattern.exec(strData)) {
        // Get the delimiter that was found.
        var strMatchedDelimiter = arrMatches[1];
        // Check to see if the given delimiter has a length
        // (is not the start of string) and if it matches
        // field delimiter. If id does not, then we know
        // that this delimiter is a row delimiter.
        if (strMatchedDelimiter.length && (strMatchedDelimiter != strDelimiter)) {
            // Since we have reached a new row of data,
            // add an empty row to our data array.
            arrData.push([]);
        }
        // Now that we have our delimiter out of the way,
        // let's check to see which kind of value we
        // captured (quoted or unquoted).
        if (arrMatches[2]) {
            // We found a quoted value. When we capture
            // this value, unescape any double quotes.
            var strMatchedValue = arrMatches[2].replace(
            new RegExp("\"\"", "g"), "\"");
        } else {
            // We found a non-quoted value.
            var strMatchedValue = arrMatches[3];
        }
        // Now that we have our value string, let's add
        // it to the data array.
        arrData[arrData.length - 1].push(strMatchedValue);
    }
    // Return the parsed data.
    return (arrData.slice(0,arrData.length-1));
}




function CSV2JSON(csv) {
    var array = CSVToArray(csv);
    var objArray = [];
    
    
    for (var i = 1; i < array.length; i++) {
        objArray[i - 1] = {};
        for (var k = 0; k < array[0].length && k < array[i].length; k++) {
            var key = array[0][k];
            objArray[i - 1][key] = array[i][k]
        }
    }
    var addData={}
    addData["data"]=objArray;
    return (addData);
}





function transpose(a) {

  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }

  return t;
}

function numInstanceOrig(arr,index) {
  console.log('numinstance');
  console.log(arr);
  var a = [],
    b = [],
    prev;
  //console.log('arrlength');
  //console.log(arr[0].length);
  //arr.sort();
  for (var i = 0; i < arr[index].length; i++) {
    //if (arr[i] !== prev) {
      //var temp=String(arr[i]).split(",")[index];
      
      //console.log("postSplit");
      //console.log(temp2);
      console.log("index",index);
      //if(exclusive.includes(temp2)){
	      if (temp2== "NA"){
	      	//console.log("an NA");
	      	a.push("0.0");
	      }
	      else{
	      	a.push(temp2);
	      }
      //}
      
      }
    
  	b.push(0);
  console.log('apost');
  console.log(a);

  return [a, b];
}

function numInstance(arr,index,exclusive) {
  console.log('numinstance');
  console.log(arr);
  var a = [],
    b = [],
    prev;
  //console.log('arrlength');
  //console.log(arr[0].length);
  //arr.sort();
  for (var i = 0; i < arr[0].length; i++) {
    //if (arr[i] !== prev) {
      //var temp=String(arr[i]).split(",")[index];
      var t1=arr[0][i];
      //console.log('t1',t1);
      var temp=String(t1);
      //console.log("befSplit",temp);
      var temp2=temp.split(",")[index];
      //console.log("postSplit");
      //console.log(temp2);
      console.log("index",index);
      if(exclusive.includes(temp.split(",")[0])){
	      if (temp2== "NA"){
	      	//console.log("an NA");
	      	a.push("0.0");
	      }
	      else{
	      	a.push(temp2);
	      }
      }
      
      }
    
  	b.push(0);
  console.log('apost');
  console.log(a);

  return [a, b];
}


function transpose(a) {
  console.log('transpose');
  console.log(a);
  // Calculate the width and height of the Array
  var w = a.length || 0;
  var h = a[0] instanceof Array ? a[0].length : 0;

  // In case it is a zero matrix, no transpose routine needed.
  if(h === 0 || w === 0) { return []; }

  /**
   * @var {Number} i Counter
   * @var {Number} j Counter
   * @var {Array} t Transposed data is stored in this array.
   */
  var i, j, t = [];

  // Loop through every item in the outer array (height)
  for(i=0; i<h; i++) {

    // Insert a new row (array)
    t[i] = [];

    // Loop through every item per item in outer array (width)
    for(j=0; j<w; j++) {

      // Save transposed data.
      t[i][j] = a[j][i];
    }
  }
  console.log('posttranspose');
  console.log(t);

  return t;
}



function tsvJSON(tsv){

  var lines=tsv.split("\n");

  var result = [];
  console.log('FIRSTLINE',lines[0]);
  // NOTE: If your columns contain commas in their values, you'll need
  // to deal with those before doing the next step 
  // (you might convert them to &&& or something, then covert them back later)
  // jsfiddle showing the issue https://jsfiddle.net/
  var headers=lines[0].split("\t");

  for(var i=1;i<lines.length;i++){

      var obj = {};
      var currentline=lines[i].split("\t");

      for(var j=0;j<headers.length;j++){
      	  if(headers[j]=='rf_predict_prob'){
      	    if(currentline[j]=='0'){
      	     obj[headers[j]]='0.00';
      	    }
      	    else{
      	    obj[headers[j]] = currentline[j];
      	    }
      	  }
      	  else{
          obj[headers[j]] = currentline[j];
          }
      }

      result.push(obj);

  }

  //return result; //JavaScript object
  return result; //JSON
}

function base64Encode(str) {
        var CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
        var out = "", i = 0, len = str.length, c1, c2, c3;
        while (i < len) {
            c1 = str.charCodeAt(i++) & 0xff;
            if (i == len) {
                out += CHARS.charAt(c1 >> 2);
                out += CHARS.charAt((c1 & 0x3) << 4);
                out += "==";
                break;
            }
            c2 = str.charCodeAt(i++);
            if (i == len) {
                out += CHARS.charAt(c1 >> 2);
                out += CHARS.charAt(((c1 & 0x3)<< 4) | ((c2 & 0xF0) >> 4));
                out += CHARS.charAt((c2 & 0xF) << 2);
                out += "=";
                break;
            }
            c3 = str.charCodeAt(i++);
            out += CHARS.charAt(c1 >> 2);
            out += CHARS.charAt(((c1 & 0x3) << 4) | ((c2 & 0xF0) >> 4));
            out += CHARS.charAt(((c2 & 0xF) << 2) | ((c3 & 0xC0) >> 6));
            out += CHARS.charAt(c3 & 0x3F);
        }
        return out;
    }


var a="{{request}}";
//console.log('a');
console.log('befdatafile',a);
b=a.split("?=")[1].split("&")[0];
console.log('datafile',b);
var data_file=b;
var data_file2=data_file.split("?")[1];



//var data_file2=data_file.split("?")[1];
//var imgName=a.split("?=")[1].split("&")[1];
//console.log('pname',imgName);


//$(document).ready(function() {
	//document.getElementById("seqlogo").src="";
	var dataout;
	var currcantype='';
	$.ajax({
	    url: 'https://immuneportal.ccbb.iupui.edu/api/experiments/'+data_file,
	    method: 'GET',
	    //datatype: 'json',
	    
	    
       	    //console.log(data_file);
            //console.log(data_file);
	    success: function (data) {
	    	//console.log('experiment info',data['experimentOutputs']);
	    	console.log(data);
	    	
	    	
	    	for(var i=0; i<data['experimentInputs'].length;i++){
	    		//console.log(Object.keys(data['experimentInputs'][i]))
	    		if(data['experimentInputs'][i]['name']=='Cancer Type'){
	    			currcantype=data['experimentInputs'][i]['value']
	    			console.log(currcantype);
	    		}
	    	}
	    	//SET THE CANCER TITLES ACCORDINGLY 
                document.getElementById('hlaLoadHeader').innerHTML="HLA Load (Load Based on Individual's <b>"+currcantype+"</b> Sample)"
                document.getElementById('relGenExpHeader').innerHTML="Gene Expression Relative to TCGA for cancer type <b>"+currcantype+"</b>"
                
	    	for(var i=0; i<data['experimentOutputs'].length;i++){
	    		if(data['experimentOutputs'][i]['name']=='Associated Files'){
	    			console.log('files',data['experimentOutputs'][i]['value']);
	    			var allFiles=data['experimentOutputs'][i]['value'].split(',');
	    			//console.log('#files',data['experimentOutputs'][i]['value'].split(',').length);
	    			///api/download?data-product-uri=airavata-dp%3A%2F%2Fddc9b4c5-7704-421a-a458-81a717203c6b
	    			
	    			console.log('EXPOUT',data['experimentOutputs']);
	    			var geneFreq='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[1];
	    			//var finalOutput='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[4];
	    			var tableData='https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+allFiles[0];
	    			console.log('TABLEDATA',tableData);
	    			
	    			//FOR GENEFREQ SECTION
	    			
	    			$.ajax({
				    url: geneFreq,
				    method: 'GET',
				    success: function (geneFreqData) {
	    				     //console.log(CSVToArray(geneFreqData))
					      
					      //TITLES OF CSV COLUMNS
						console.log(CSVToArray(geneFreqData)[0]);
						var temp=CSVToArray(geneFreqData);
						var geneFreqDataTitles=temp[0];
						var geneFreqDataArr=temp.slice(1, temp.length);
						var personalGenesDict={};
						console.log(geneFreqDataArr);
						for(var i=0;i<geneFreqDataArr.length;i++){
							

							personalGenesDict[geneFreqDataArr[i][0]]=parseFloat(geneFreqDataArr[i][1])
						}
						console.log('personGenes',personalGenesDict);
						$.ajax({
							    async: false,
							    url: 'https://immuneportal.ccbb.iupui.edu/static/files/OSF_TCGA_ALL_ICB_genes_exprs.csv',
							    method: 'GET',
							    //datatype: 'json',
							    
							    
						       	    //console.log(data_file);
							    //console.log(data_file);
							    success: function (tcgaGeneFreqs) {
							    		var temp=CSVToArray(tcgaGeneFreqs);
							    		//console.log(tcgaGeneFreqsArray);
									var tcgaGeneFreqsTitles=temp[0];
									console.log(tcgaGeneFreqsTitles);
									var tcgaGeneFreqsArr=temp.slice(1, temp.length);
									console.log(tcgaGeneFreqsArr[0]);
									
									var titleArr=tcgaGeneFreqsTitles[0].split(',');
									var geneDict={};
									//INITIALIZING GENE DICT
									for(var j=0;j<titleArr.length;j++){

										if(j==0| j==titleArr.length-1){
											continue;
										}
										else{
											geneDict[titleArr[j]]=[];
										}
										
									}
									//FILLING GENE DICT
									for(var i=0;i<tcgaGeneFreqsArr.length;i++){
										var currLst=tcgaGeneFreqsArr[i][0].split(',');
										if(currLst[titleArr.length-1]==currcantype){
											
											for(var j=0;j<titleArr.length;j++){

												if(j==0| j==titleArr.length-1){
													continue;
												}
												else{
													geneDict[titleArr[j]].push(parseFloat(currLst[j]));
												}
												
											}
										}
										else{
											continue;
										}
									}
									console.log('part of b2m',geneDict['B2M'].slice(0,5));
									
									var keysForIt=Object.keys(geneDict).reverse();
									var traces=[];
									for(var i=0;i<keysForIt.length;i++){
										var temptrace={
											x: geneDict[keysForIt[i]],
											marker: {color: '#000000'},
									  		type: 'box',
									  		name: keysForIt[i]
										}
										traces.push(temptrace);
										temptrace={
											x: [personalGenesDict[keysForIt[i]]],
											y: [keysForIt[i]],
											marker: {
												color: '#FF0000',
												/*line:{
													width:6
												}*/
											},
									  		type: 'scatter',
									  		//text: 'Patient '+keysForIt[i],
									  		name: 'Patient '+keysForIt[i]
										}
										traces.push(temptrace);
									}
							    		//TITLES OF CSV COLUMNS
							    		/*console.log(dataout[0]);
						    			var trace1 = {
									  x: [1, 2, 3, 4, 4, 4, 8, 9, 10],
									  type: 'box',
									  name: 'Set 1'
									};

									var trace2 = {
									  x: [2, 3, 3, 3, 3, 5, 6, 6, 7],
									  type: 'box',
									  name: 'Set 2'
									};*/

									var data = traces;

									var layout = {
									  title: 'Relative Gene Concentrations'
									};

									Plotly.newPlot('boxes', data, layout);
								}
						});
				
				
	    				}
	    			});
	    			/*$(document).ready(function () {
					$(".upset2").blowup({
						background : "#FCEBB6"
					});
				})*/
	    			//FOR GENEFREQ Section ENDED
	    			$.ajax({
				    async: false,
				    url: tableData,
				    //url: tableData,
				    method: 'GET',
				    //datatype: 'json',
				    
				    
			       	    //console.log(data_file);
				    //console.log(data_file);
				    success: function (data) {
				//console.log('hello');
						//console.log(data_file);
						    //console.log(data);
						    var dataout=CSVToArray(data);
						    dataoutTable = dataout.slice(1, dataout.length);
						    
						    var jsonout=tsvJSON(data);
						    jsonout.pop();
						    console.log('JSONOUT',jsonout);
						
						    //console.log('postfunc',dataout);
						    //$.fn.dataTableExt.sErrMode = 'throw';
						    /*var base=document.getElementById('forResultTable');
						    var tbl=document.createElement('table');
						    tbl.id='resultTable'
						    tbl.classList.add('table');
						    //tbl.classList.add('table-striped');
						    tbl.classList.add('table-bordered');
						    tbl.classList.add('table-hover');
						    
						    //tbl.set="table table-striped table-bordered table-hover";
						    tbl.style="cursor:pointer;"
						    tbl.cellspacing="0";
						    tbl.width="100%";
						    var thd=document.createElement('thead');
						    var tr=document.createElement('tr');
						    */
						    var columnDefs=[]
						    console.log('DataOut',dataout[0]);
						    for(var i=0;i<dataout[0].length;i++){
						    	//var th=document.createElement('th');
						    	//var tempspan;
						    	if(dataout[0][i]!= 'peptide' & dataout[0][i]!='geneSymbol' & dataout[0][i]!='rf_predict'){
						    		//th.style="display:none";
						    		columnDefs.push({"targets":[i],"visible":false,"searchable":false});

						    	}
						    	else{
						    		columnDefs.push({"targets":[i],"visible":true,"searchable":true,"className":"noVis"})
						    	}

						    	
						    	
						    }
						    
						    
					/*	    
						    <table id="resultTable" class="table table-striped table-bordered table-hover" style="cursor:pointer;" cellspacing="0" width="100%">
				  <thead>
				      <tr>
					   <th style="display:none">id</th>
					    <th class="HlaWithComment">hla <span class="HlaComment"  >human leukocyte antigen (HLA) allele </span> </th>
					    
					    <th class="WtWithComment" >mut_pep<span class="WtComment">peptide derived from retained intron regions; </span></th>
					    <th style="display:none" class="WtWithComment">wt_pep <span class="WtComment">WildType Peptide: </span></th>
					    <th style="display:none">pep_hydro</th>
					    <th style="display:none">pep_polar</th>
					    <th style="display:none">pep_charge</th>
					    <th style="display:none">pep_entropy</th>
					    <th style="display:none">pep_molsize</th>
					    <th style="display:none">pep_TAP</th>
					    <th style="display:none">pep_Cle</th>
					    <th style="display:none">pep_Comb</th>
					    <th style="display:none">neo_aff</th>
					    <th style="display:none">neo_rank</th>
					    <th style="display:none">iedb_immunescore</th>
					    <th style="display:none">pep_DAI</th>
					    <th style="display:none">pep_Rscore</th>
					    <th style="display:none">pep_NRP</th>
					    <th style="display:none"> pep_pTuneos</th>
					    <th>rf_predict</th>
				      </tr>
				  </thead>
				</table>
				*/
				


descriptionDict={'Id': 'random assigned peptide id','hla':  'human leukocyte antigen (HLA) allele','mut_pep': 'peptide derived from retained intron regions','wt_pep': 'the most similar peptide from human reference protein database','pep_hydro': 'sum of amino acids hydrophobicity for peptide','pep_polar': 'sum of amino acids residues polarity for peptide','pep_charge': 'sum of amino acids residues net charge for peptide (at neutral pH)','pep_entropy': 'sum of amino acids residues entropy for peptide','pep_molsize': 'sum of amino acids residues molecular weight for peptide','pep_TAP':  'TAP transport efficiency, predicted by NetCTL1.2','pep_Cle': 'C terminal cleavage affinity, predicted by NetCTL1.2','pep_Comb': 'Overall prediction score, predicted by NetCTL1.2','neo_aff': 'NetMHCpan 4.1 predicted IC50 values (Aff, nM unit)','neo_rank': 'NetMHCpan 4.1 predicted percentile rank binding affinity value','iedb_immunescore': 'Class I Immunogenicity score from IEDB, http://tools.iedb.org/immunogenicity','pep_DAI': 'differential agretopicity index, from Ghorani et al’s study (PMID:29361136)','pep_Rscore': 'neoantigen fitness score from Łuksza et al’s study (PMID:29132144)','pep_NRP': 'Neoantigen Recognition Potential score, Balachandran et al’s study (PMID:29132146)','pep_pTuneos': 'pTuneos prediction score, Zhou et al’s study (PMID:31666118)','rf_predict': 'intronNeoantigen random forest model prediction, Positive/Negative represent the ability for recognition by CD8+ cytotoxic T cells' };	
var customTitleFormatter = function(cell){

    var headerEl = cell.getElement(); //get the column header element from above the title element

    headerEl.addEventListener("mouseover", function(e){
        //do something
        console.log('mousover');
        //console.log(e);
        console.log(headerEl.innerText);
        
        document.getElementById('colDescription').innerHTML=headerEl.innerText+':'+descriptionDict[header.innerText];
        
        
    });

    return cell.getValue(); // ensure the title is set on the cell
}	
				
var minMaxFilterEditor = function(cell, onRendered, success, cancel, editorParams){

    
    
    
    

    var container = document.createElement("span");
    
    var subcontainer=document.createElement("div");

    var start=document.createElement('input');
    start.type="range";
    start.id="start";
    start.setAttribute('class','sliderinput');
    start.min='0.00';
    start.max='1.00';
    start.setAttribute("step","0.01");
    start.setAttribute("value",'0.00');
    start.style.width='120px;'
    
    var end=document.createElement('input');
    end.type="range";
    end.id="end";
    end.setAttribute('class','sliderinput');
    end.min='0.00';
    end.max='1.00';
    end.setAttribute("step","0.01");
    end.setAttribute("value",'1.00');
    end.style.width='120px;'
/*
    var start=document.createElement('input');
    start.setAttribute('class','slider');
    start.setAttribute("oninput","document.getElementById('price-min-lbl').innerHTML = 'Min:'+this.value");
    start.type='range';
    start.name='price-min';
    start.id='price-minid';
    start.setAttribute("value",'0');
    start.setAttribute("step","0.01");
    start.min='0.00';
    start.max='1.00';
    
    
    var end=document.createElement('input');
    end.setAttribute('class','slider');
    end.setAttribute("oninput","document.getElementById('price-max-lbl').innerHTML = ' Max:' +this.value");
    end.type='range';
    end.name='price-max';
    end.id='price-maxid';
    end.setAttribute("value",'1.00');
    end.setAttribute("step","0.01");
    end.min='0.00';
    end.max='1.00';
    
    */
    var fieldvalue = "<label for='price-min' id='price-min-lbl'>0</label>100</label>";
    
    //subcontainer.innerHTML=fieldvalue;
   
    //  


    
    
    
    
    var startbox = document.createElement("input");
    startbox.setAttribute("type", "number");
    startbox.setAttribute("placeholder", "Min");
    startbox.setAttribute('class','minbox');
    startbox.setAttribute("min", 0.00);
    startbox.setAttribute("max", 1.00);
    startbox.style.margin="0px";
    startbox.style.left="0px";
    startbox.setAttribute("step", 0.01);
    //startbox.setAttribute("value", 0.00);
    startbox.style.width = "70px";
    startbox.style.zIndex="2";

    startbox.value = cell.getValue();
    
    
    var endbox = document.createElement("input");
    endbox.setAttribute("type", "number");
    endbox.setAttribute("placeholder", "Max");
    endbox.setAttribute('class','maxbox');
    endbox.setAttribute("min", 0.00);
    endbox.setAttribute("step", 0.01);
    endbox.setAttribute("max", 1.00);
    //endbox.setAttribute("value", 1.00);
    //endbox.style.padding = "4px";
    endbox.style.width = "70px";
    endbox.style.zIndex="2";
    //endbox.style.boxSizing = "border-box";

    endbox.value = cell.getValue();

    function buildValues(){
    	//startbox.setAttribute("value",start.value);
    	//endbox.setAttribute("value",end.value);
    	startbox.value=start.value;
    	endbox.value=end.value;
        success({
            start:start.value,
            end:end.value,
            startbox:start.value,
            endbox:end.value,
        });
    }
    
    
    var boxalter = function(eventArg) 
    {
    	start.setAttribute("value",startbox.value);
    	end.setAttribute("value",endbox.value);
        success({
            start:startbox.value,
            end:endbox.value,
        });
    }
    
    
    var MouseMove = function(eventArg) 
    {
        /* percentage position X of cursor  */
        //console.log(eventArg.clientX);
        var recta = start.getBoundingClientRect();
        var x100 = Math.abs(eventArg.clientX-recta.left)/(recta.right-recta.left);


        /* absolute distance from respective slider values */
        var da = Math.abs(start.value-x100);
        var db = Math.abs(end.value-x100);
        console.log(x100,da,db);
        console.log(recta.left);
        console.log(recta.right-recta.left);
        console.log(eventArg.clientX-recta.left);
        // Making the two sliders appear above one another only when no mouse button is pressed, this oondition may be removed at will
        if (!eventArg.buttons) 
        {
            if (da > db) 
            {
                start.style.zIndex = 1;
                end.style.zIndex = 2;
            } 
            else 
            {
                start.style.zIndex = 2;
                end.style.zIndex = 1;
            }
        }
        
        //document.querySelector('label').innerHTML = 'Red: ' + a.value + ', Green: ' + b.value + ', X: ' + eventArg.clientX;
    }
    
    
    startbox.addEventListener("change",boxalter);
    startbox.addEventListener("input",boxalter);
    endbox.addEventListener("change",boxalter);
    endbox.addEventListener("input",boxalter);
    
    start.addEventListener("mousemove",MouseMove );
    start.addEventListener("change", buildValues);
    start.addEventListener("blur", buildValues);
    //start.addEventListener("keydown", keypress);

    end.addEventListener('mousemove',MouseMove);
    end.addEventListener("change", buildValues);
    end.addEventListener("blur", buildValues);
    //end.addEventListener("keydown", keypress);
    
    
    
    container.append(startbox);
    container.append(start);
    //container.append(lmax);
    container.append(end);
    container.append(endbox);
    //container.append(lmax);

    return container;
 }
 

 /*
var minMaxFilterEditor = function(cell, onRendered, success, cancel, editorParams){

    var end;
    
    
    var container = document.createElement("span");

    //create and style inputs
    var start = document.createElement("input");
    start.setAttribute("type", "number");
    start.setAttribute("placeholder", "Min");
    start.setAttribute("min", 0);
    start.setAttribute("max", 100);
    start.style.padding = "4px";
    start.style.width = "50%";
    start.style.boxSizing = "border-box";

    start.value = cell.getValue();

    function buildValues(){
        success({
            start:start.value,
            end:end.value,
        });
    }

    function keypress(e){
        if(e.keyCode == 13){
            buildValues();
        }

        if(e.keyCode == 27){
            cancel();
        }
    }

    end = start.cloneNode();
    end.setAttribute("placeholder", "Max");

    start.addEventListener("change", buildValues);
    start.addEventListener("blur", buildValues);
    start.addEventListener("keydown", keypress);

    end.addEventListener("change", buildValues);
    end.addEventListener("blur", buildValues);
    end.addEventListener("keydown", keypress);


    container.appendChild(start);
    container.appendChild(end);

    return container;
 }*/

//custom max min filter function
function minMaxFilterFunction(headerValue, rowValue, rowData, filterParams){
    //headerValue - the value of the header filter element
    //rowValue - the value of the column in this row
    //rowData - the data for the row being filtered
    //filterParams - params object passed to the headerFilterFuncParams property
        if(rowValue){
            //console.log(rowValue)
            //if (rowValue==0){
            //	console.log('zero case');
            //}
            //console.log(headerValue.start);
            if((headerValue.start != 0 || headerValue.start != 0.0)  && headerValue.start != "" ){
                //console.log('start is 0');
                if((headerValue.end != 1 || headerValue.start != 1.0) && headerValue.end != ""){
                //console.log('end is 1');
                //console.log('filtering');
                    return rowValue >= headerValue.start && rowValue <= headerValue.end &&  rowValue != 0 && rowValue != "0" && rowValue != 0.00 && rowValue != 1 && rowValue != 1.00;
                }
                if(headerValue.end != ""){
                    return rowValue >= headerValue.start && rowValue <= headerValue.end && rowValue != 0 && rowValue != "0" && rowValue != 0.00 ;
                }
               	else{
                    return rowValue >=headerValue.start && rowValue != 0 && rowValue != "0" && rowValue != 0.00;
                }
            }
            if( headerValue.start != "" ){
                if((headerValue.end != 1 || headerValue.start != 1.0) && headerValue.end != ""){
                //console.log('filtering');
                    return rowValue >= headerValue.start && rowValue <= headerValue.end && rowValue != 1 && rowValue != 1.00;
                }
                if(headerValue.end != ""){
                    return rowValue >= headerValue.start && rowValue <= headerValue.end ;
                }
               	else{
                    return rowValue >=headerValue.start ;
                }
            }
        }

    return true; //must return a boolean, true if it passes the filter.
}


var browser =  new Browser({
								      chr:          '22',
								      viewStart:    30700000,
								      viewEnd:      30900000,

								      coordSystem: {
									speciesName: 'Human',
									taxon: 9606,
									auth: 'GRCh',
									version: '38',
									ucscName: 'hg38'
								      },

								      sources:     [{name:                 'Genome',
										     twoBitURI:            'https://www.biodalliance.org/datasets/hg38.2bit',
										     tier_type:            'sequence'},
										    {name:                 'Genes',
										     desc:                 'Gene structures from GENCODE 20',
										     bwgURI:               'https://www.biodalliance.org/datasets/gencode.bb',
										     stylesheet_uri:       'https://www.biodalliance.org/stylesheets/gencode.xml',
										     collapseSuperGroups:  true,
										     trixURI:              'https://www.biodalliance.org/datasets/geneIndex.ix'},
										    {name:                 'Repeats',
										     desc:                 'Repeat annotation from Ensembl',
										     bwgURI:               'https://www.biodalliance.org/datasets/repeats.bb',
										     stylesheet_uri:       'https://www.biodalliance.org/stylesheets/bb-repeats.xml'},
										    {name:                 'Conservation',
										     desc:                 'Conservation', 
										     bwgURI:               'https://hgdownload.cse.ucsc.edu/goldenpath/hg38/phyloP100way/hg38.phyloP100way.bw',
										     noDownsample:         true}],

								    });



var retJson=[{
  "peptide": "AEMKVWET",
  "gene": "ENSG00000180096-16@chr16:30379535-30379931",
  "tpm": "0.8473",
  "HLA-A*24:02": "0.016",
  "HLA-A*25:01": "ENSG00000180096",
  "HLA-B*07:02": "SEPTIN1",
  "HLA-B*18:01": "52.0",
  "HLA-C*07:02": "55.0",
  "HLA-C*12:03": "50.0",
  "hla": "1.825",
  "wt_pep": "53.0",
  "pep_hydro": "56.667",
  "pep_polar": "HLA-B18:01",
  "pep_charge": "AEQGVWET",
  "pep_entropy": "-4.6",
  "pep_molsize": "53.56",
  "pep_TAP": "-1",
  "pep_Cle": "1754.17",
  "pep_Comb": "992",
  "neo_aff": "-0.296",
  "neo_rank": "0.09325",
  "iedb_immunescore": "0.29258",
  "pep_DAI": "5301.04",
  "pep_Rscore": "1.825",
  "pep_NRP": "0.0592",
  "pep_pTuneos": "-0.9346",
  "rf_predict": "0.0",
  "rf_predict_prob": "-0.0",
  "ensembl": "-0.0",
  "geneSymbol": "Negative",
  "undefined": "-0.0"
}];


function onHoverTable(){
	console.log('columnHovered');

    }

var table2 = new Tabulator("#example-table", {
    data:jsonout,
    height:"311px",
    layout:"fitDataFill",
    //movableColumns: true,
    columns:[
        {title:"peptide", field:"peptide", width:150,headerFilter:"input",titleFormatter:customTitleFormatter,tooltip:onHoverTable},
        {title:"gene", field:"gene", width:150,headerFilter:"input"},
        {title:"tpm", field:"tpm", width:150,headerFilter:"input"},
        {title:"rf_predict_prob", field:"rf_predict_prob", width:300,headerFilter:minMaxFilterEditor, headerFilterFunc:minMaxFilterFunction, headerFilterLiveFilter:false},
        {title:"ensembl", field:"ensembl", width:150,headerFilter:"input"},
        {title:"geneSymbol", field:"geneSymbol", width:150,headerFilter:"input"},
        {title:"HLA-A*24:02", field:"HLA-A*24:02", width:150,visible:false,headerFilter:"input"},
        {title:"HLA-A*25:01", field:"HLA-A*25:01", width:150,visible:false,headerFilter:"input"},
        {title:"HLA-B*07:02", field:"HLA-B*07:02", width:150,visible:false,headerFilter:"input"},
        {title:"HLA-B*18:01", field:"HLA-B*18:01", width:150,visible:false,headerFilter:"input"},
        {title:"HLA-C*07:02", field:"HLA-C*07:02", width:150,visible:false,headerFilter:"input"},
        {title:"HLA-C*12:03", field:"HLA-C*12:03", width:150,visible:false,headerFilter:"input"},
        {title:"HLA", field:"hla", width:150,visible:false,headerFilter:"input"},
        {title:"wt_pep", field:"wt_pep", width:150,visible:false,headerFilter:"input"},
        {title:"pep_hydro", field:"pep_hydro", width:150,visible:false,headerFilter:"input"},
        {title:"pep_polar", field:"pep_polar", width:150,visible:false,headerFilter:"input"},
        {title:"pep_charge", field:"pep_charge", width:150,visible:false,headerFilter:"input"},
        {title:"pep_entropy", field:"pep_entropy", width:150,visible:false,headerFilter:"input"},
        {title:"pep_molsize", field:"pep_molsize", width:150,visible:false,headerFilter:"input"},
        {title:"pep_TAP", field:"pep_TAP", width:150,visible:false,headerFilter:"input"},
        {title:"pep_Cle", field:"pep_Cle", width:150,visible:false,headerFilter:"input"},
        {title:"pep_Comb", field:"pep_Comb", width:150,visible:false,headerFilter:"input"},
        {title:"neo_aff", field:"neo_aff", width:150,visible:false,headerFilter:"input"},
        {title:"neo_rank", field:"neo_rank", width:150,visible:false,headerFilter:"input"},
        {title:"iedb_immunescore", field:"iedb_immunescore", width:150,visible:false,headerFilter:"input"},
        {title:"pep_DAI", field:"pep_DAI", width:150,visible:false,headerFilter:"input"},
        {title:"pep_Rscore", field:"pep_Rscore", width:150,visible:false,headerFilter:"input"},
        {title:"pep_NRP", field:"pep_NRP", width:150,visible:false,headerFilter:"input"},
        {title:"pep_pTuneos", field:"pep_pTuneos", width:150,visible:false,headerFilter:"input"},
        {title:"rf_predict", field:"rf_predict", width:150,visible:false,headerFilter:"input"},
        {title:"pep_pTuneos", field:"pep_pTuneos", width:150,visible:false,headerFilter:"input"},
        
        ],
        dataFiltered: function(filters, rows) {
        var el = document.getElementById("search_count");
        el.innerHTML = rows.length;
        //var el = document.getElementById("total_count");
        //el.innerHTML = data.length;
    },
    rowClick:function(e, row){
    //e - the click event object
    //row - row component
    console.log('rowclick',row);
    
    var headers=[];
    for(var i=0;i<Object.keys(jsonout[0]).length;i++){
    	headers.push(Object.keys(jsonout[0])[i]);
    }
    console.log('rowclickheaders',headers);
    var rowvals=row._row.cells;
    //var retJson={};
    var tempDict={};
    retJson=[];
    for(var i=0;i<rowvals.length;i++){
    	tempDict[headers[i]]=rowvals[i]['value'];
    }
    retJson.push(tempDict);
    console.log('retJson',retJson);
    
    document.getElementById('rowData').style.display="block";
     document.getElementById('example-table3').style.display="block";
    
    var temp=row._row.cells[1].value.split('@')[1]
  var chr=temp.split(':')[0].substring(3)
  var start=parseInt(temp.split(':')[1].split('-')[0])
  var end=parseInt(temp.split(':')[1].split('-')[1])
   browser.setLocation(chr, start-50, end + 50);
   document.getElementById('gblabel').style.display="block";
  document.getElementById('svgHolder').style.display="block";
    },
   
});



var table3 = new Tabulator("#example-table3", {
    data:retJson,
    height:"80px",
    layout:"fitDataFill",
    //movableColumns: true,
    columns:[
        {title:"peptide", field:"peptide", width:150},
        {title:"gene", field:"gene", width:150},
        {title:"tpm", field:"tpm", width:150},
        {title:"rf_predict_prob", field:"rf_predict_prob", width:150},
        {title:"ensembl", field:"ensembl", width:150},
        {title:"geneSymbol", field:"geneSymbol", width:150},
        {title:"HLA-A*24:02", field:"HLA-A*24:02", width:150,visible:true},
        {title:"HLA-A*25:01", field:"HLA-A*25:01", width:150,visible:true},
        {title:"HLA-B*07:02", field:"HLA-B*07:02", width:150,visible:true},
        {title:"HLA-B*18:01", field:"HLA-B*18:01", width:150,visible:true},
        {title:"HLA-C*07:02", field:"HLA-C*07:02", width:150,visible:true},
        {title:"HLA-C*12:03", field:"HLA-C*12:03", width:150,visible:true},
        {title:"HLA", field:"hla", width:150,visible:true},
        {title:"wt_pep", field:"wt_pep", width:150,visible:true},
        {title:"pep_hydro", field:"pep_hydro", width:150,visible:true},
        {title:"pep_polar", field:"pep_polar", width:150,visible:true},
        {title:"pep_charge", field:"pep_charge", width:150,visible:true},
        {title:"pep_entropy", field:"pep_entropy", width:150,visible:true},
        {title:"pep_molsize", field:"pep_molsize", width:150,visible:true},
        {title:"pep_TAP", field:"pep_TAP", width:150,visible:true},
        {title:"pep_Cle", field:"pep_Cle", width:150,visible:true},
        {title:"pep_Comb", field:"pep_Comb", width:150,visible:true},
        {title:"neo_aff", field:"neo_aff", width:150,visible:true},
        {title:"neo_rank", field:"neo_rank", width:150,visible:true},
        {title:"iedb_immunescore", field:"iedb_immunescore", width:150,visible:true},
        {title:"pep_DAI", field:"pep_DAI", width:150,visible:true},
        {title:"pep_Rscore", field:"pep_Rscore", width:150,visible:true},
        {title:"pep_NRP", field:"pep_NRP", width:150,visible:true},
        {title:"pep_pTuneos", field:"pep_pTuneos", width:150,visible:true},
        {title:"rf_predict", field:"rf_predict", width:150,visible:true},
        {title:"pep_pTuneos", field:"pep_pTuneos", width:150,visible:true},
        
        ],
        dataFiltered: function(filters, rows) {
        var el = document.getElementById("search_count");
        el.innerHTML = rows.length;
        //var el = document.getElementById("total_count");
        //el.innerHTML = data.length;
    },
    
   
});






//var total_count = $(".tabulator-footer").find('.tabulator-cell:first-child()').text();
//$("#total_count").text(total_count);

$( "#colShow" ).change(function() {
  //alert( "Handler for .change() called." );
  var x=$("#colShow" ).val();
  console.log(x);
  //console.log(table2);
  //table2.showColumn(x);
  var objkeys=Object.entries(table2.columnManager.columnsByField);
  for(var i=0;i<objkeys.length;i++){
  	var title=objkeys[i][1].definition['field'];
  	//console.log('title',title,'x',x);
  	if(title==x){
  		console.log(table2.columnManager);
  		var visible=table2.columnManager.columnsByField[title].definition.visible;
  		console.log(visible);
  		if(visible==true || visible==null){
  			//table2.hideColumn(x);
  			//console.log('hiding');
  			table2.columnManager.columnsByField[title].definition.visible=false;
  			table2.hideColumn(x);
  		}
  		else{
  		        table2.columnManager.columnsByField[title].definition.visible=true;
  			table2.showColumn(x);
  			console.log('showing');
  		}
  	}
  	
  }
});


document.getElementById("download-csv").addEventListener("click", function(){
    table2.download("csv", "data.csv");
});



				
						   console.log('DATATOTABLE',dataoutTable);
						    
						    
							    //SEARCHABLE COLUMNS	  
							
							//END OF SEARCHABLE COLUMNS 
							    
							
							   	  
								
							document.getElementById('additionalInfo').style.display="block";
							//document.getElementById('additionalInfo2').style.display="block";
							document.getElementById('outer').style.display="block";
							
							
							
								

					    /*
					    //HLALOAD SECTION
					    $.ajax({
					    async: false,
					    url: 'https://immuneportal.ccbb.iupui.edu/static/files/TCGA_IRneoAg_load_in_8cancers_summary.csv',
					    method: 'GET',
					    //datatype: 'json',
					    
					    
					       	    //console.log(data_file);
						    //console.log(data_file);
						    success: function (TCGAbreakdown) {
						    		var TCGAarray=CSVToArray(TCGAbreakdown);
						    		console.log(TCGAarray);
						    		//var TCGAjson=CSV2JSON(TCGAbreakdown);
						    		//console.log('tcgajson',TCGAjson);
						    		var currCanc=[];
						    		for(var i=1;i<TCGAarray.length;i++){
						    			if(TCGAarray[i][0].includes(currcantype)){
						    				currCanc.push(parseInt(TCGAarray[i][0].split(',')[1]));
						    			}
						    		}
						    		console.log('currcanc',currCanc);
						    		console.log('CURRCANCE Min:',Math.min.apply(Math,currCanc));
						    		console.log('CURRCANCE RANGE:',Math.max.apply(Math,currCanc)-Math.min.apply(Math,currCanc));
						    		var currCancRange=Math.max.apply(Math,currCanc)-Math.min.apply(Math,currCanc)
						    		
								  
								  
								

  


								  
							}
					    });
					    
					    */
					    
					    
					    
					    
					    
					    
					    $.ajax({
					   async: false,
					   type: "GET",
					    url: "https://immuneportal.ccbb.iupui.edu/static/files/1000genomes_HLA_freq_statistics.csv",
					    

					    success: function (data) {
					    	var genomes=CSVToArray(data);
					    	//console.log('made it here');
					    	
						//console.log('json',data);
							var userspecific=dataout;
							console.log('userspecific',userspecific);
							console.log('indexof',userspecific[0].indexOf('hla'));
							console.log('transposedus',transpose(userspecific.slice(1,userspecific.length)));
							var allTypes=transpose(userspecific.slice(1,userspecific.length))[userspecific[0].indexOf('hla')]
							console.log('allTypes',allTypes);
							console.log('genomes',transpose(genomes.slice(1,genomes.length)));
							var count = {};
							allTypes.forEach(e => count[e] ? count[e]++ : count[e] = 1 );
							console.log("count");
							console.log(count);
							
							var nKeys=Object.values(count).length;
							
							var values=Object.values(count);
							console.log('thesevals',values);
							var proportions=new Array(nKeys);
							var total=Object.values(count).reduce((a, b) => a + b, 0);
							for (i=0; i<nKeys; i++){
							     proportions[i]=values[i]/total;
							}
							console.log("Individual HLA",Object.keys(count));
							//count.entries();
							//var hlatypes= numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('HLA_list'))[0];
							
							//console.log(transpose(genomes.slice(1,genomes.length)));
							//console.log('hlalist');
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length)),[genomes[0].indexOf('HLA_list')])[0]);
							//console.log('all');
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length)),[genomes[0].indexOf('ALL')])[0]);
							//console.log(genomes.slice(0))
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length))[genomes[0].indexOf('')])[0]
							//console.log(numInstance(transpose(genomes.slice(1,genomes.length))[genomes[0].indexOf('')])[0]
							//var hlatypes= numInstanceOrig(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('HLA_list'))[0];
							
							//console.log("indexTest",String(genomes[0]).split(",").indexOf('HLA_list'))
							console.log('HOPEFULLY KEYS',Object.keys(count));
							console.log('numinstance',numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('SAS'),Object.keys(count)));
							
							/*var trace1 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('ALL'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'ALL' 
							};*/

							var trace2 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('AFR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'AFR' 
							};

							var trace3 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('AMR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'AMR' 
							};

							var trace4 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('EAS'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'EAS' 
							};
							var trace5 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('EUR'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'EUR' 
							};

							var trace6 = {
							  x: Object.keys(count),
							  y: numInstance(transpose(genomes.slice(1,genomes.length)),String(genomes[0]).split(",").indexOf('SAS'),Object.keys(count))[0],
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'SAS' 
							};
							/*
							var trace7 = {
							  x: Object.keys(count),
							  y: proportions,
							  xaxis: 'HLA Type',
							  yaxis: 'Proportion',
							  type: 'bar',
							  name: 'User Sample' 
							};*/

							//var data = [ trace2, trace3, trace4,trace5,trace6,trace7];
							var data = [ trace2, trace3, trace4,trace5,trace6];

							var layout = {
							  //grid: {rows: 1, columns: 2, pattern: 'independent'},
							  autosize: true,
							  title: {text:"1000 Genomes HLA Proportions"},
							  xaxis: {title: { text:'HLA subtype'}},
							  yaxis: {title: { text:'Proportion'}},
							  margin: {
							    l: 50,
							    r: 50,
							    b: 100,
							    t: 100,
							    pad: 4
							  },
										  
							  //width: 1800,
					 		  //height: 1250,
							};



							Plotly.newPlot('myDiv1', data, layout);
						
					    }
					  });
					    
					    
					  	
					}
				});
				
				

			
			
			
			$.ajax({
			   async: true,
			   type: "GET",
			   cache: true ,
			    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/upset_view",
			    contentType: "image/jpg",
			    data:tableData.split('=')[1],
			    success: function (data) {
				//console.log('json',data);
				console.log(data);
				//console.log('https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+data)
				document.getElementById('upset').src = 'data:image/jpg;base64,' + data;
				
			//document.getElementById('hlaload').src = 'data:image/jpg;base64,' + +dataUris[0];
				   
				
			    
				
			    }
			});
			
			
			
			$.ajax({
			   async: true,
			   type: "GET",
			   cache: true,
			    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/hlaload_view",
			    mimeType: "text/plain; charset=x-user-defined",
			    //contentType: "image/jpg",
			    data:tableData.split('=')[1],
			    success: function (laterdata) {
				//console.log('json',data);
				console.log(laterdata);
				hlaload=document.getElementById('hlaLoad');
				image=document.createElement('img');
				image.setAttribute('src', 'data:image/jpg;base64,'+base64Encode(laterdata));
				hlaload.appendChild(image);
				//console.log('https://immuneportal.ccbb.iupui.edu/api/download?data-product-uri='+data)
				//document.getElementById('hlaLoad').src = 'data:image/jpg;base64,' + base64Encode(laterdata);
			//document.getElementById('hlaload').src = 'data:image/jpg;base64,' + +dataUris[0];

			    }
			});
			
			
	
	
    
    //console.log('firstOnly',output[0]);
    //$.fn.dataTable.ext.errMode = 'throw';
    
    /*$.each(data, function(i, data) {
	    var body = "<tr>";
	    body    += "<td>" + data.id + "</td>";
	    body    += "<td>" + data.hla + "</td>";
	    body    += "<td>" + data.mut_pep + "</td>";
	    body    += "<td>" + data.wt_pep + "</td>";
	    body    += "<td>" + data.pep_hydro + "</td>";
	    body    += "<td>" + data.pep_polar + "</td>";
	    body    += "<td>" + data.pep_charge + "</td>";
	    body    += "<td>" + data.pep_entropy + "</td>";
	    body    += "<td>" + data.pep_molsize + "</td>";
	    body    += "<td>" + data.pep_TAP + "</td>";
	    body    += "<td>" + data.pep_Cle + "</td>";
	    body    += "<td>" + data.pep_Comb + "</td>";
	    body    += "<td>" + data.neo_aff + "</td>";
	    body    += "<td>" + data.neo_rank + "</td>";
	    body    += "<td>" + data.iedb_immunescore + "</td>";
	    body    += "<td>" + data.pep_DAI + "</td>";
	    body    += "<td>" + data.pep_Rscore + "</td>";
	    body    += "<td>" + data.pep_NRP + "</td>";
	    body    += "<td>" + data.pep_pTuneos + "</td>";
	    body    += "<td>" + data.rf_predict + "</td>";
	    
	    body    += "</tr>";
	    $( '#resultTable').append(body);
    });*/
    
    
    
    
    
  

	    		
	    		break;
	    		}
	    	}
	    }
	});
	//console.log('https://immuneportal.ccbb.iupui.edu/api/experiments/'+data_file);
	
	
	
	    
    
    //console.log('firstOnly',output[0]);
    //$.fn.dataTable.ext.errMode = 'throw';
    
    /*$.each(data, function(i, data) {
	    var body = "<tr>";
	    body    += "<td>" + data.id + "</td>";
	    body    += "<td>" + data.hla + "</td>";
	    body    += "<td>" + data.mut_pep + "</td>";
	    body    += "<td>" + data.wt_pep + "</td>";
	    body    += "<td>" + data.pep_hydro + "</td>";
	    body    += "<td>" + data.pep_polar + "</td>";
	    body    += "<td>" + data.pep_charge + "</td>";
	    body    += "<td>" + data.pep_entropy + "</td>";
	    body    += "<td>" + data.pep_molsize + "</td>";
	    body    += "<td>" + data.pep_TAP + "</td>";
	    body    += "<td>" + data.pep_Cle + "</td>";
	    body    += "<td>" + data.pep_Comb + "</td>";
	    body    += "<td>" + data.neo_aff + "</td>";
	    body    += "<td>" + data.neo_rank + "</td>";
	    body    += "<td>" + data.iedb_immunescore + "</td>";
	    body    += "<td>" + data.pep_DAI + "</td>";
	    body    += "<td>" + data.pep_Rscore + "</td>";
	    body    += "<td>" + data.pep_NRP + "</td>";
	    body    += "<td>" + data.pep_pTuneos + "</td>";
	    body    += "<td>" + data.rf_predict + "</td>";
	    
	    body    += "</tr>";
	    $( '#resultTable').append(body);
    });*/
    
    
    
    
	/*
    $.ajax({
	   type: "GET",
	    url: "https://immuneportal.ccbb.iupui.edu/immuneportal_django_app/image_view/?"+data_file2,
	    contentType: "image/jpg",

	    success: function (data) {
		//console.log('json',data);
		console.log('made it');
		document.getElementById('seqlogo').src = 'data:image/jpg;base64,' + data;
	    }
	});
	*/
	
	
    //});

	
    
	
   
	    	
	    
/*

	    $.fn.dataTable.ext.errMode = 'throw';
	    $('#resultTable').DataTable( {
		    "processing": true,
		"scrollX": true,
		"data":output,
	    } );
	    var table = $('#resultTable').DataTable();*/
/*
var data = [{
  type: 'table',
  header: {
    values: output[0],
    align: "center",
    line: {width: 1, color: 'black'},
    fill: {color: "grey"},
    font: {family: "Arial", size: 12, color: "white"}
  },
  cells: {
    values: transpose(output.slice(1,output.length)),
    align: "center",
    line: {color: "black", width: 1},
    font: {family: "Arial", size: 11, color: ["black"]}
  }
}]

Plotly.newPlot('myPltTable', data);

//CLICK EVENTS FOR PLOT TO UPDATE TABLE

myPlot.on('plotly_click', function(data){
    console.log(data);
    
    var newData=[];
    var curveNumber=data.points[0].curveNumber;
    for (i=1;i<output.length;i++){
    	//probability curve
    	if(curveNumber==0){
    		if(output[i].includes(data.points[0].y)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==1){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    	else if(curveNumber==2){
    		if(output[i].includes(data.points[0].x)){
    			newData.push(output[i]);
    		}
    	}
    }
    
    var data = [{
	  type: 'table',
	  header: {
	    values: output[0],
	    align: "center",
	    line: {width: 1, color: 'black'},
	    fill: {color: "grey"},
	    font: {family: "Arial", size: 12, color: "white"}
	  },
	  cells: {
	    values: transpose(newData.slice(0,newData.length)),
	    align: "center",
	    line: {color: "black", width: 1},
	    font: {family: "Arial", size: 11, color: ["black"]}
	  }
	}]
    //document.getElementById("myPltTable").setAttribute('data',data);
    //Plotly.redraw('myPltTable');
    Plotly.newPlot('myPltTable', data);

    
    /*var pts = '';
    for(var i=0; i < data.points.length; i++){
        pts = 'x = '+data.points[i].x +'\ny = '+
            data.points[i].y.toPrecision(4) + '\n\n';
    }
    alert('Closest point clicked:\n\n'+pts);
});*/







</script>
<script>
    
    const { models, services, session, utils } = AiravataAPI;
    
    

    const appInterfaceId = "Echo_23d67491-1bef-47bd-a0f5-faf069e09773";

</script>

{% endblock scripts %}
